diff --git a/srchybrid/UPnPImplMiniLib.cpp b/srchybrid/UPnPImplMiniLib.cpp
index 9c41ae1..561c94a 100644
--- a/srchybrid/UPnPImplMiniLib.cpp
+++ b/srchybrid/UPnPImplMiniLib.cpp
@@ -21,9 +21,9 @@
 #include "UPnPImplMiniLib.h"
 #include "Log.h"
 #include "Otherfunctions.h"
-#include "miniupnpc\miniupnpc.h"
-#include "miniupnpc\upnpcommands.h"
-#include "miniupnpc\upnperrors.h"
+#include "miniupnpc\include\miniupnpc.h"
+#include "miniupnpc\include\upnpcommands.h"
+#include "miniupnpc\include\upnperrors.h"
 #include "opcodes.h"
 
 
