diff --git a/srchybrid/ColourPopup.h b/srchybrid/ColourPopup.h
index cdd1058..5fd8050 100644
--- a/srchybrid/ColourPopup.h
+++ b/srchybrid/ColourPopup.h
@@ -25,7 +25,7 @@ class CColourPicker;
 struct ColourTableEntry
 {
 	COLORREF crColour;
-	TCHAR	 *szName;
+	LPCTSTR	 szName;
 };
 
 
diff --git a/srchybrid/EMSocket.cpp b/srchybrid/EMSocket.cpp
index 975aec0..749f251 100644
--- a/srchybrid/EMSocket.cpp
+++ b/srchybrid/EMSocket.cpp
@@ -36,7 +36,7 @@ static char THIS_FILE[] = __FILE__;
 
 namespace
 {
-	inline void EMTrace(char *fmt, ...)
+	inline void EMTrace(const char *fmt, ...)
 	{
 #ifdef EMSOCKET_DEBUG
 		va_list argptr;
@@ -912,7 +912,7 @@ int CEMSocket::Receive(void *lpBuf, int nBufLen, int nFlags)
 		}
 		break;
 	case SOCKET_ERROR:
-		char *p = NULL;
+		const char *p = NULL;
 		switch (CAsyncSocket::GetLastError()) {
 		case WSANOTINITIALISED:
 			ASSERT(0);
diff --git a/srchybrid/MetaDataDlg.cpp b/srchybrid/MetaDataDlg.cpp
index 3824709..f4ab80c 100644
--- a/srchybrid/MetaDataDlg.cpp
+++ b/srchybrid/MetaDataDlg.cpp
@@ -434,7 +434,7 @@ void CMetaDataDlg::RefreshData()
 				lvi.iItem = iItem;
 
 				// intentionally left blank as it's not a real meta tag
-				lvi.pszText = _T("");
+				lvi.pszText = const_cast<LPTSTR>(_T(""));
 				lvi.iSubItem = META_DATA_COL_TYPE;
 				m_tags.SetItem(&lvi);
 
diff --git a/srchybrid/OtherFunctions.cpp b/srchybrid/OtherFunctions.cpp
index ebb6f16..5189e35 100644
--- a/srchybrid/OtherFunctions.cpp
+++ b/srchybrid/OtherFunctions.cpp
@@ -2180,8 +2180,8 @@ CString GetFormatedUInt(ULONG ulVal)
 		nf.LeadingZero = 0;
 		nf.Grouping = 3;
 		// we are hardcoding the following two format chars intentionally because the C-RTL also has the decimal sep hardcoded to '.'
-		nf.lpDecimalSep = _T(".");
-		nf.lpThousandSep = _T(",");
+		nf.lpDecimalSep = const_cast<LPTSTR>(_T("."));
+		nf.lpThousandSep = const_cast<LPTSTR>(_T(","));
 		nf.NegativeOrder = 0;
 	}
 	CString strVal;
@@ -2202,8 +2202,8 @@ CString GetFormatedUInt64(ULONGLONG ullVal)
 		nf.LeadingZero = 0;
 		nf.Grouping = 3;
 		// we are hardcoding the following two format chars intentionally because the C-RTL also has the decimal sep hardcoded to '.'
-		nf.lpDecimalSep = _T(".");
-		nf.lpThousandSep = _T(",");
+		nf.lpDecimalSep = const_cast<LPTSTR>(_T("."));
+		nf.lpThousandSep = const_cast<LPTSTR>(_T(","));
 		nf.NegativeOrder = 0;
 	}
 	const int iBuffSize = _countof(szVal) * 2;
diff --git a/srchybrid/PPgGeneral.cpp b/srchybrid/PPgGeneral.cpp
index ee026e2..a127a94 100644
--- a/srchybrid/PPgGeneral.cpp
+++ b/srchybrid/PPgGeneral.cpp
@@ -118,7 +118,7 @@ BOOL CPPgGeneral::OnInitDialog()
 	thePrefs.GetLanguages(aLanguageIDs);
 	for (INT_PTR i = 0; i < aLanguageIDs.GetCount(); ++i) {
 		TCHAR szLang[128];
-		TCHAR *pLang = szLang;
+		LPCTSTR pLang = szLang;
 		int ret = GetLocaleInfo(aLanguageIDs[i], LOCALE_SLANGUAGE, szLang, _countof(szLang));
 
 		if (ret == 0)
diff --git a/srchybrid/SearchListCtrl.cpp b/srchybrid/SearchListCtrl.cpp
index 4fa5238..f529d03 100644
--- a/srchybrid/SearchListCtrl.cpp
+++ b/srchybrid/SearchListCtrl.cpp
@@ -1687,8 +1687,8 @@ CString CSearchListCtrl::FormatFileSize(ULONGLONG ullFileSize) const
 			nf.LeadingZero = 1;
 			nf.Grouping = 3;
 			// we are hardcoding the following two format chars by intention because the C-RTL also has the decimal sep hardcoded to '.'
-			nf.lpDecimalSep = _T(".");
-			nf.lpThousandSep = _T(",");
+			nf.lpDecimalSep = const_cast<LPTSTR>(_T("."));
+			nf.lpThousandSep = const_cast<LPTSTR>(_T(","));
 			nf.NegativeOrder = 0;
 		}
 		CString sVal, strVal;
diff --git a/srchybrid/SendMail.cpp b/srchybrid/SendMail.cpp
index 61a79d7..b4650cf 100644
--- a/srchybrid/SendMail.cpp
+++ b/srchybrid/SendMail.cpp
@@ -56,7 +56,7 @@ void LogCertificate(PCCERT_CONTEXT pCertContext)
 		if (CertNameToStr(X509_ASN_ENCODING | PKCS_7_ASN_ENCODING, &pCertInfo->Issuer, CERT_X500_NAME_STR, szString, _countof(szString)))
 			DebugLog(LOG_DONTNOTIFY, _T("Email Encryption: Issuer: %s"), szString);
 	} else {
-		if (CertGetNameString(pCertContext, CERT_NAME_SIMPLE_DISPLAY_TYPE, CERT_NAME_DISABLE_IE4_UTF8_FLAG, szOID_COMMON_NAME, szString, _countof(szString)))
+		if (CertGetNameString(pCertContext, CERT_NAME_SIMPLE_DISPLAY_TYPE, CERT_NAME_DISABLE_IE4_UTF8_FLAG, const_cast<LPSTR>(szOID_COMMON_NAME), szString, _countof(szString)))
 			DebugLog(LOG_DONTNOTIFY, _T("Email Encryption: Name: %s"), szString);
 
 		BYTE md5[16];
@@ -97,7 +97,7 @@ bool Encrypt(const CStringA &rstrContentA, CByteArray &raEncrypted, LPCWSTR pwsz
 			//PCCERT_CONTEXT RecipientCertArray[1] = { pRecipientCert };
 
 			CRYPT_ALGORITHM_IDENTIFIER EncryptAlgorithm = {};
-			EncryptAlgorithm.pszObjId = szOID_RSA_DES_EDE3_CBC;
+			EncryptAlgorithm.pszObjId = const_cast<LPSTR>(szOID_RSA_DES_EDE3_CBC);
 
 			CRYPT_ENCRYPT_MESSAGE_PARA EncryptParams = {};
 			EncryptParams.cbSize = (DWORD)sizeof EncryptParams;
diff --git a/srchybrid/UPnPImplWinServ.cpp b/srchybrid/UPnPImplWinServ.cpp
index 91c5bf6..545c796 100644
--- a/srchybrid/UPnPImplWinServ.cpp
+++ b/srchybrid/UPnPImplWinServ.cpp
@@ -1180,7 +1180,7 @@ CString translateUPnPResult(HRESULT hr)
 {
 	if (hr >= UPNP_E_ACTION_SPECIFIC_BASE && hr <= UPNP_E_ACTION_SPECIFIC_MAX)
 		return CString(_T("Action Specific Error"));
-	TCHAR *p;
+	LPCTSTR p;
 	switch (hr) {
 	case UPNP_E_ROOT_ELEMENT_EXPECTED:
 		p = _T("Root Element Expected");
