diff --git a/srchybrid/ClientList.cpp b/srchybrid/ClientList.cpp
index 782c84a..624708d 100644
--- a/srchybrid/ClientList.cpp
+++ b/srchybrid/ClientList.cpp
@@ -745,7 +745,7 @@ bool CClientList::IncomingBuddy(const Kademlia::CContact *contact, const Kademli
 	CUpDownClient *pNewClient = new CUpDownClient(NULL, contact->GetTCPPort(), contact->GetIPAddress(), 0, 0, false);
 	pNewClient->SetKadPort(contact->GetUDPPort());
 	pNewClient->SetKadState(KS_INCOMING_BUDDY);
-	byte ID[sizeof Kademlia::CUInt128];
+	byte ID[sizeof(Kademlia::CUInt128)];
 	contact->GetClientID().ToByteArray(ID);
 	pNewClient->SetUserHash(ID); //??
 	buddyID.ToByteArray(ID);
diff --git a/srchybrid/QArray.h b/srchybrid/QArray.h
index 20a6c2f..b7c575d 100644
--- a/srchybrid/QArray.h
+++ b/srchybrid/QArray.h
@@ -115,6 +115,6 @@ public:
 //
 template <class T, class TP> void CQArray<T, TP>::QuickSort(BOOL bAscending/* = TRUE*/)
 {
-	if (GetSize() > 1)
-		::QuickSort(GetData(), (int)GetSize(), bAscending);
+	if (this->GetSize() > 1)
+		::QuickSort(this->GetData(), (int)this->GetSize(), bAscending);
 }
\ No newline at end of file
diff --git a/srchybrid/Ring.h b/srchybrid/Ring.h
index ee31987..4664cb1 100644
--- a/srchybrid/Ring.h
+++ b/srchybrid/Ring.h
@@ -95,10 +95,10 @@ void CRing<TYPE>::SetBuffer(UINT_PTR nSize)
 	TYPE *dst = new TYPE[nSize];
 	if (m_nCount)
 		if (m_pHead > m_pTail) {
-			memcpy(dst, m_pHead, (m_pEnd - m_pHead) * sizeof TYPE);
-			memcpy(&dst[m_pEnd - m_pHead], m_pData, (m_pTail - m_pData + 1) * sizeof TYPE);
+			memcpy(dst, m_pHead, (m_pEnd - m_pHead) * sizeof(TYPE));
+			memcpy(&dst[m_pEnd - m_pHead], m_pData, (m_pTail - m_pData + 1) * sizeof(TYPE));
 		} else
-			memcpy(dst, m_pHead, (m_pTail - m_pHead + 1) * sizeof TYPE);
+			memcpy(dst, m_pHead, (m_pTail - m_pHead + 1) * sizeof(TYPE));
 	delete[] m_pData;
 	m_nSize = nSize;
 	m_pHead = m_pData = dst;
