diff --git a/srchybrid/EncryptedStreamSocket.cpp b/srchybrid/EncryptedStreamSocket.cpp
index b28d4eb..f68fcb6 100644
--- a/srchybrid/EncryptedStreamSocket.cpp
+++ b/srchybrid/EncryptedStreamSocket.cpp
@@ -411,7 +411,7 @@ void CEncryptedStreamSocket::StartNegotiation(bool bOutgoing)
 		ASSERT(m_cryptDHA.MinEncodedSize() <= DHAGREEMENT_A_BITS / 8);
 		CryptoPP::Integer cryptDHPrime((byte*)dh768_p, PRIMESIZE_BYTES);  // our fixed prime
 		// calculate g^a % p
-		CryptoPP::Integer cryptDHGexpAmodP = CryptoPP::a_exp_b_mod_c(CryptoPP::Integer(2), m_cryptDHA, cryptDHPrime);
+		CryptoPP::Integer cryptDHGexpAmodP = a_exp_b_mod_c(CryptoPP::Integer(2), m_cryptDHA, cryptDHPrime);
 		ASSERT(m_cryptDHA.MinEncodedSize() <= PRIMESIZE_BYTES);
 		// put the result into a buffer
 		uchar aBuffer[PRIMESIZE_BYTES];
@@ -575,7 +575,7 @@ int CEncryptedStreamSocket::Negotiate(const uchar *pBuffer, int nLen)
 					m_pfiReceiveBuffer->Read(aBuffer, PRIMESIZE_BYTES);
 					CryptoPP::Integer cryptDHAnswer(static_cast<byte*>(aBuffer), PRIMESIZE_BYTES);
 					CryptoPP::Integer cryptDHPrime(static_cast<byte*>(dh768_p), PRIMESIZE_BYTES);  // our fixed prime
-					CryptoPP::Integer cryptResult = CryptoPP::a_exp_b_mod_c(cryptDHAnswer, m_cryptDHA, cryptDHPrime);
+					CryptoPP::Integer cryptResult = a_exp_b_mod_c(cryptDHAnswer, m_cryptDHA, cryptDHPrime);
 
 					m_cryptDHA = 0;
 					DEBUG_ONLY(memset(aBuffer, 0, sizeof aBuffer));
diff --git a/srchybrid/FileInfoDialog.cpp b/srchybrid/FileInfoDialog.cpp
index af4802a..709a812 100644
--- a/srchybrid/FileInfoDialog.cpp
+++ b/srchybrid/FileInfoDialog.cpp
@@ -1785,7 +1785,7 @@ bool CGetMediaInfoThread::GetMediaInfo(HWND hWndOwner, const CShareableFile *pFi
 														bFoundHeader = true;
 													} else {
 														mi->strInfo << _T("   ") << GetResString(IDS_CODEC) << _T(":\t") << GetVideoFormatName(pVIH->bmiHeader.biCompression) << _T("\n");
-														mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << abs(pVIH->bmiHeader.biWidth) << _T(" x ") << abs(pVIH->bmiHeader.biHeight) << _T("\n");
+														mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << (int)abs(pVIH->bmiHeader.biWidth) << _T(" x ") << (int)abs(pVIH->bmiHeader.biHeight) << _T("\n");
 														// do not use that 'dwBitRate', whatever this number is, it's not
 														// the bit rate of the *encoded* video stream. Seems to be the bit rate
 														// of the *decoded* stream
@@ -1906,7 +1906,7 @@ bool CGetMediaInfoThread::GetMediaInfo(HWND hWndOwner, const CShareableFile *pFi
 												mi->strInfo << _T("\n");
 											mi->OutputFileName();
 											mi->strInfo.SetSelectionCharFormat(mi->strInfo.m_cfBold);
-											mi->strInfo << GetResString(IDS_UNKNOWN) << _T(" Stream #") << i + 1 << _T("\n");
+											mi->strInfo << GetResString(IDS_UNKNOWN) << _T(" Stream #") << (int)(i + 1) << _T("\n");
 
 											double fLength;
 											if (SUCCEEDED(pMediaDet->get_StreamLength(&fLength)) && fLength) {
diff --git a/srchybrid/KadLookupGraph.cpp b/srchybrid/KadLookupGraph.cpp
index 6c1ae14..1f54938 100644
--- a/srchybrid/KadLookupGraph.cpp
+++ b/srchybrid/KadLookupGraph.cpp
@@ -351,7 +351,7 @@ void CKadLookupGraph::OnPaint()
 							} else
 								pen = &gdipPenGray;
 
-							gdipGraphic.DrawLine(pen, pFrom.x, pFrom.y, pointTo.x, pointTo.y);
+							gdipGraphic.DrawLine(pen, (INT)pFrom.x, (INT)pFrom.y, (INT)pointTo.x, (INT)pointTo.y);
 						}
 					}
 				}
diff --git a/srchybrid/MediaInfo.cpp b/srchybrid/MediaInfo.cpp
index 7658b8f..37f33a7 100644
--- a/srchybrid/MediaInfo.cpp
+++ b/srchybrid/MediaInfo.cpp
@@ -798,7 +798,7 @@ bool GetRIFFHeaders(LPCTSTR pszFileName, SMediaInfo *mi, bool &rbIsAVI, bool bFu
 									if (strmhdr.fmt.bmi && strmhdr.dwFormatLen >= sizeof *strmhdr.fmt.bmi) {
 										mi->strInfo << _T("   ") << GetResString(IDS_CODEC) << _T(":\t") << GetVideoFormatName(strmhdr.fmt.bmi->biCompression) << _T("\n");
 										if (strmhdr.fmt.bmi->biWidth && strmhdr.fmt.bmi->biHeight) {
-											mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << abs(strmhdr.fmt.bmi->biWidth) << _T(" x ") << abs(strmhdr.fmt.bmi->biHeight) << _T("\n");
+											mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << (int)abs(strmhdr.fmt.bmi->biWidth) << _T(" x ") << (int)abs(strmhdr.fmt.bmi->biHeight) << _T("\n");
 											float fAspectRatio = fabsf((float)strmhdr.fmt.bmi->biWidth / (float)strmhdr.fmt.bmi->biHeight);
 											mi->strInfo << _T("   ") << GetResString(IDS_ASPECTRATIO) << _T(":\t") << fAspectRatio << _T("  (") << GetKnownAspectRatioDisplayString(fAspectRatio) << _T(")\n");
 										}
@@ -2252,7 +2252,7 @@ bool GetWMHeaders(LPCTSTR pszFileName, SMediaInfo *mi, bool &rbIsWM, bool bFullI
 											mi->strInfo << _T("   ") << GetResString(IDS_CODEC) << _T(":\t") << GetVideoFormatName(pVideoInfo->bmiHeader.biCompression) << _T("\n");
 											if (pVideoInfo->dwBitRate)
 												mi->strInfo << _T("   ") << GetResString(IDS_BITRATE) << _T(":\t") << (UINT)((pVideoInfo->dwBitRate + 500) / 1000) << _T(" kbit/s\n");
-											mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << abs(pVideoInfo->bmiHeader.biWidth) << _T(" x ") << abs(pVideoInfo->bmiHeader.biHeight) << _T("\n");
+											mi->strInfo << _T("   ") << GetResString(IDS_WIDTH) << _T(" x ") << GetResString(IDS_HEIGHT) << _T(":\t") << (int)abs(pVideoInfo->bmiHeader.biWidth) << _T(" x ") << (int)abs(pVideoInfo->bmiHeader.biHeight) << _T("\n");
 											float fAspectRatio = fabsf((float)pVideoInfo->bmiHeader.biWidth / (float)pVideoInfo->bmiHeader.biHeight);
 											mi->strInfo << _T("   ") << GetResString(IDS_ASPECTRATIO) << _T(":\t") << fAspectRatio << _T("  (") << GetKnownAspectRatioDisplayString(fAspectRatio) << _T(")\n");
 
