diff --git a/srchybrid/PPgConnection.cpp b/srchybrid/PPgConnection.cpp
index 83d08c7..e1c5703 100644
--- a/srchybrid/PPgConnection.cpp
+++ b/srchybrid/PPgConnection.cpp
@@ -191,9 +191,6 @@ void CPPgConnection::LoadSettings()
 		GetDlgItem(IDC_NETWORK_KADEMLIA)->EnableWindow(thePrefs.GetUDPPort() > 0);
 		CheckDlgButton(IDC_NETWORK_ED2K, static_cast<UINT>(thePrefs.networked2k));
 
-		WORD wv = thePrefs.GetWindowsVersion();
-		GetDlgItem(IDC_PREF_UPNPONSTART)->EnableWindow(wv != _WINVER_95_ && wv != _WINVER_98_ && wv != _WINVER_NT4_);
-
 		CheckDlgButton(IDC_PREF_UPNPONSTART, static_cast<UINT>(thePrefs.IsUPnPEnabled()));
 
 		//ShowLimitValues(); - will be called in OnLimiterChange()
diff --git a/srchybrid/PPgGeneral.cpp b/srchybrid/PPgGeneral.cpp
index a127a94..b3cca64 100644
--- a/srchybrid/PPgGeneral.cpp
+++ b/srchybrid/PPgGeneral.cpp
@@ -95,12 +95,7 @@ void CPPgGeneral::LoadSettings()
 	CheckDlgButton(IDC_STARTMIN, static_cast<UINT>(thePrefs.startMinimized));
 	CheckDlgButton(IDC_STARTWIN, static_cast<UINT>(thePrefs.m_bAutoStart));
 
-	if (thePrefs.GetWindowsVersion() != _WINVER_95_)
-		CheckDlgButton(IDC_PREVENTSTANDBY, static_cast<UINT>(thePrefs.GetPreventStandby()));
-	else {
-		CheckDlgButton(IDC_PREVENTSTANDBY, 0);
-		GetDlgItem(IDC_PREVENTSTANDBY)->EnableWindow(FALSE);
-	}
+	CheckDlgButton(IDC_PREVENTSTANDBY, static_cast<UINT>(thePrefs.GetPreventStandby()));
 
 	CString strBuffer;
 	strBuffer.Format(_T("%u %s"), thePrefs.versioncheckdays, (LPCTSTR)GetResString(IDS_DAYS2));
