diff --git a/srchybrid/PPgTweaks.cpp b/srchybrid/PPgTweaks.cpp
index d1b207b..9ad2d8e 100644
--- a/srchybrid/PPgTweaks.cpp
+++ b/srchybrid/PPgTweaks.cpp
@@ -325,7 +325,6 @@ void CPPgTweaks::DoDataExchange(CDataExchange *pDX)
 	// File related group
 	//
 	DDX_TreeCheck(pDX, IDC_EXT_OPTS, m_htiSparsePartFiles, m_bSparsePartFiles);
-	m_ctrlTreeOptions.SetCheckBoxEnable(m_htiSparsePartFiles, wv != _WINVER_VISTA_ /*only disable on Vista, not later versions*/);
 	DDX_TreeCheck(pDX, IDC_EXT_OPTS, m_htiImportParts, m_bImportParts);
 	DDX_TreeCheck(pDX, IDC_EXT_OPTS, m_htiFullAlloc, m_bFullAlloc);
 	DDX_TreeCheck(pDX, IDC_EXT_OPTS, m_htiCheckDiskspace, m_bCheckDiskspace);
diff --git a/srchybrid/Preferences.cpp b/srchybrid/Preferences.cpp
index bff29af..7269139 100644
--- a/srchybrid/Preferences.cpp
+++ b/srchybrid/Preferences.cpp
@@ -3044,18 +3044,7 @@ void CPreferences::ChangeUserDirMode(int nNewMode)
 
 bool CPreferences::GetSparsePartFiles()
 {
-	// Vista's Sparse File implementation seems to be buggy as far as i can see
-	// If a sparse file exceeds a given limit of write I/O operations in a certain order
-	// (or i.e. end to beginning) in its lifetime, it will at some point throw out
-	// a FILE_SYSTEM_LIMITATION error and deny any writing to this file.
-	// It was suggested that Vista might limit the data runs, which would lead to such behaviour,
-	// but wouldn't make much sense for a sparse file implementation nevertheless.
-	// Due to the fact that eMule writes a lot of small blocks into sparse files and flushes them
-	// every 6 seconds, this problem pops up sooner or later for all big files.
-	// I don't see any way to walk around this for now
-	// Update: This problem seems to be fixed on Win7, possibly on earlier Vista ServicePacks too
-	//		   In any case, we allow sparse files for versions earlier and later than Vista
-	return m_bSparsePartFiles && (GetWindowsVersion() != _WINVER_VISTA_);
+	return m_bSparsePartFiles;
 }
 
 bool CPreferences::IsRunningAeroGlassTheme()
