diff --git a/srchybrid/OtherFunctions.cpp b/srchybrid/OtherFunctions.cpp
index e6ce323..fc2a7ad 100644
--- a/srchybrid/OtherFunctions.cpp
+++ b/srchybrid/OtherFunctions.cpp
@@ -618,60 +618,6 @@ UINT GetMaxWindowsTCPConnections()
 	return UNLIMITED;
 }
 
-#pragma warning(push)
-#pragma warning(disable:4996) //GetVersionEx()
-WORD DetectWinVersion()
-{
-	OSVERSIONINFOEX osvi;
-	osvi.dwOSVersionInfoSize = (DWORD)sizeof(OSVERSIONINFOEX);
-
-	if (!GetVersionEx((OSVERSIONINFO*)&osvi)) {
-		osvi.dwOSVersionInfoSize = (DWORD)sizeof(OSVERSIONINFO);
-		if (!GetVersionEx((OSVERSIONINFO*)&osvi))
-			return 0;
-	}
-
-	switch (osvi.dwPlatformId) {
-	case VER_PLATFORM_WIN32_NT:
-		if (osvi.dwMajorVersion <= 4)
-			return _WINVER_NT4_;
-		if (osvi.dwMajorVersion == 5) {
-			if (osvi.dwMinorVersion == 0)
-				return _WINVER_2K_;
-			if (osvi.dwMinorVersion == 1)
-				return _WINVER_XP_;
-			if (osvi.dwMinorVersion == 2)
-				return _WINVER_2003_;
-		}
-		if (osvi.dwMajorVersion == 6) {
-			if (osvi.dwMinorVersion == 0)
-				return _WINVER_VISTA_;
-			if (osvi.dwMinorVersion == 1)
-				return _WINVER_7_;
-			if (osvi.dwMinorVersion == 2)
-				return _WINVER_8_;
-			if (osvi.dwMinorVersion == 3)
-				return _WINVER_8_1_;
-		}
-		if (osvi.dwMajorVersion == 10 && osvi.dwMinorVersion == 0)
-			return _WINVER_10_; //for Windows 11 - osvi.dwBuildNumber>=22000
-		return _WINVER_7_; // never return Win95 if we get the info about an NT system
-
-	case VER_PLATFORM_WIN32_WINDOWS:
-		if (osvi.dwMajorVersion == 4) {
-			if (osvi.dwMinorVersion == 0)
-				return _WINVER_95_;
-			if (osvi.dwMinorVersion == 10)
-				return _WINVER_98_;
-			if (osvi.dwMinorVersion == 90)
-				return _WINVER_ME_;
-		}
-	}
-
-	return _WINVER_95_;		// there shouldn't be anything lower than this
-}
-#pragma warning(pop)
-
 uint64 GetFreeDiskSpaceX(LPCTSTR pDirectory)
 {
 	ULARGE_INTEGER nFreeDiskSpace;
diff --git a/srchybrid/OtherFunctions.h b/srchybrid/OtherFunctions.h
index f9b31a3..4cc8d71 100644
--- a/srchybrid/OtherFunctions.h
+++ b/srchybrid/OtherFunctions.h
@@ -317,21 +317,6 @@ ULONGLONG GetModuleVersion(HMODULE hModule);
 
 UINT GetMaxWindowsTCPConnections();
 
-#define _WINVER_95_		0x0400	// 4.0
-#define _WINVER_NT4_	0x0401	// 4.1 (baked version)
-#define _WINVER_98_		0x040A	// 4.10
-#define _WINVER_ME_		0x045A	// 4.90
-#define _WINVER_2K_		0x0500	// 5.0
-#define _WINVER_XP_		0x0501	// 5.1
-#define _WINVER_2003_	0x0502	// 5.2
-#define _WINVER_VISTA_	0x0600	// 6.0
-#define _WINVER_7_		0x0601	// 6.1
-#define	_WINVER_S2008_	0x0601	// 6.1
-#define _WINVER_8_		0x0602	// 6.2
-#define _WINVER_8_1_	0x0603	// 6.3
-#define _WINVER_10_		0x0a00	// 10.0
-
-WORD		DetectWinVersion();
 uint64		GetFreeDiskSpaceX(LPCTSTR pDirectory);
 ULONGLONG	GetDiskFileSize(LPCTSTR pszFilePath);
 int			GetAppImageListColorFlag();
diff --git a/srchybrid/Preferences.cpp b/srchybrid/Preferences.cpp
index db4b581..59c374a 100644
--- a/srchybrid/Preferences.cpp
+++ b/srchybrid/Preferences.cpp
@@ -443,7 +443,6 @@ CIni	*CPreferences::skinsIni;
 Preferences_Ext_Struct *CPreferences::prefsExt;
 CArray<Category_Struct*> CPreferences::catArr;
 UINT	CPreferences::m_nWebMirrorAlertLevel;
-WORD	CPreferences::m_wWinVer;
 bool	CPreferences::m_bUseOldTimeRemaining;
 
 int		CPreferences::m_byLogLevel;
@@ -2395,16 +2394,6 @@ void CPreferences::LoadPreferences()
 	SetLanguage();
 }
 
-WORD CPreferences::GetWindowsVersion()
-{
-	static bool bWinVerAlreadyDetected = false;
-	if (!bWinVerAlreadyDetected) {
-		bWinVerAlreadyDetected = true;
-		m_wWinVer = DetectWinVersion();
-	}
-	return m_wWinVer;
-}
-
 //////////////////////////////////////////////////////////
 // category implementations
 //////////////////////////////////////////////////////////
diff --git a/srchybrid/Preferences.h b/srchybrid/Preferences.h
index fd77eb6..003ee2c 100644
--- a/srchybrid/Preferences.h
+++ b/srchybrid/Preferences.h
@@ -1053,7 +1053,6 @@ public:
 	static bool		GetMessageEnableSmileys()			{ return m_bMessageEnableSmileys; }
 	static bool		GetIRCEnableUTF8()					{ return m_bIRCEnableUTF8; }
 
-	static WORD		GetWindowsVersion();
 	static bool		IsRunningAeroGlassTheme();
 	static bool		GetStartMinimized()					{ return startMinimized; }
 	static void		SetStartMinimized( bool instartMinimized) { startMinimized = instartMinimized; }
@@ -1404,7 +1403,6 @@ public:
 protected:
 	static CString	m_strFileCommentsFilePath;
 	static Preferences_Ext_Struct *prefsExt;
-	static WORD		m_wWinVer;
 	static CArray<Category_Struct*> catArr;
 	static CString	m_astrDefaultDirs[13];
 	static bool		m_abDefaultDirsCreated[13];
