diff --git a/srchybrid/Emule.cpp b/srchybrid/Emule.cpp
index c1d9092..e99d213 100644
--- a/srchybrid/Emule.cpp
+++ b/srchybrid/Emule.cpp
@@ -146,73 +146,62 @@ static void InitSafeSEH()
 // VS2008:	DEP can get enabled at link time by using the "DEP" option within
 //			'Visual Studio Linker Advanced Options'.
 //
-#ifndef PROCESS_DEP_ENABLE
-#define	PROCESS_DEP_ENABLE						0x00000001
-#define	PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION	0x00000002
-#endif//!PROCESS_DEP_ENABLE
-
 static void InitDEP()
 {
-	BOOL(WINAPI *pfnGetProcessDEPPolicy)(HANDLE hProcess, LPDWORD lpFlags, PBOOL lpPermanent);
-	BOOL(WINAPI *pfnSetProcessDEPPolicy)(DWORD dwFlags);
-	(FARPROC&)pfnGetProcessDEPPolicy = ::GetProcAddress(::GetModuleHandle(_T("kernel32")), "GetProcessDEPPolicy");
-	(FARPROC&)pfnSetProcessDEPPolicy = ::GetProcAddress(::GetModuleHandle(_T("kernel32")), "SetProcessDEPPolicy");
-	if (pfnGetProcessDEPPolicy && pfnSetProcessDEPPolicy) {
-		DWORD dwFlags;
-		BOOL bPermanent;
-		if ((*pfnGetProcessDEPPolicy)(::GetCurrentProcess(), &dwFlags, &bPermanent)) {
-			// Vista SP1
-			// ===============================================================
-			//
-			// BOOT.INI nx=OptIn,  VS2003/VS2005
-			// ---------------------------------
-			// DEP flags: 00000000
-			// Permanent: 0
-			//
-			// BOOT.INI nx=OptOut, VS2003/VS2005
-			// ---------------------------------
-			// DEP flags: 00000001 (PROCESS_DEP_ENABLE)
-			// Permanent: 0
-			//
-			// BOOT.INI nx=OptIn/OptOut, VS2003 + EditBinX/NXCOMPAT
-			// ----------------------------------------------------
-			// DEP flags: 00000003 (PROCESS_DEP_ENABLE | *PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION*)
-			// Permanent: *1*
-			// ---
-			// There is no way to remove the PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION flag at runtime,
-			// because the DEP policy is already permanent due to the NXCOMPAT flag.
-			//
-			// BOOT.INI nx=OptIn/OptOut, VS2005 + /NXCOMPAT
-			// --------------------------------------------
-			// DEP flags: 00000003 (PROCESS_DEP_ENABLE | PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION)
-			// Permanent: *1*
-			//
-			// NOTE: It is ultimately important to explicitly enable the DEP policy even if the
-			// process' DEP policy is already enabled. If the DEP policy is already enabled due
-			// to an OptOut system policy, the DEP policy is though not yet permanent. As long as
-			// the DEP policy is not permanent it could get changed during runtime...
-			//
-			// So, if the DEP policy for the current process is already enabled but not permanent,
-			// it has to be explicitly enabled by calling 'SetProcessDEPPolicy' to make it permanent.
-			//
-			if (((dwFlags & PROCESS_DEP_ENABLE) == 0 || !bPermanent)
+	DWORD dwFlags;
+	BOOL bPermanent;
+	if (::GetProcessDEPPolicy(::GetCurrentProcess(), &dwFlags, &bPermanent)) {
+		// Vista SP1
+		// ===============================================================
+		//
+		// BOOT.INI nx=OptIn,  VS2003/VS2005
+		// ---------------------------------
+		// DEP flags: 00000000
+		// Permanent: 0
+		//
+		// BOOT.INI nx=OptOut, VS2003/VS2005
+		// ---------------------------------
+		// DEP flags: 00000001 (PROCESS_DEP_ENABLE)
+		// Permanent: 0
+		//
+		// BOOT.INI nx=OptIn/OptOut, VS2003 + EditBinX/NXCOMPAT
+		// ----------------------------------------------------
+		// DEP flags: 00000003 (PROCESS_DEP_ENABLE | *PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION*)
+		// Permanent: *1*
+		// ---
+		// There is no way to remove the PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION flag at runtime,
+		// because the DEP policy is already permanent due to the NXCOMPAT flag.
+		//
+		// BOOT.INI nx=OptIn/OptOut, VS2005 + /NXCOMPAT
+		// --------------------------------------------
+		// DEP flags: 00000003 (PROCESS_DEP_ENABLE | PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION)
+		// Permanent: *1*
+		//
+		// NOTE: It is ultimately important to explicitly enable the DEP policy even if the
+		// process' DEP policy is already enabled. If the DEP policy is already enabled due
+		// to an OptOut system policy, the DEP policy is though not yet permanent. As long as
+		// the DEP policy is not permanent it could get changed during runtime...
+		//
+		// So, if the DEP policy for the current process is already enabled but not permanent,
+		// it has to be explicitly enabled by calling 'SetProcessDEPPolicy' to make it permanent.
+		//
+		if (((dwFlags & PROCESS_DEP_ENABLE) == 0 || !bPermanent)
 #if _ATL_VER>0x0710
-				|| (dwFlags & PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION) == 0
+			|| (dwFlags & PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION) == 0
 #endif
-				)
-			{
-				// VS2003:	Enable DEP (with ATL-thunk emulation) if not already set by system policy
-				//			or if the policy is not yet permanent.
-				//
-				// VS2005:	Enable DEP (without ATL-thunk emulation) if not already set by system policy
-				//			or linker "/NXCOMPAT" option or if the policy is not yet permanent. We should
-				//			not reach this code path at all because the "/NXCOMPAT" option is specified.
-				//			However, the code path is here for safety reasons.
-				dwFlags = PROCESS_DEP_ENABLE;
-				// VS2005: Disable ATL thunks.
-				dwFlags |= PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION;
-				(*pfnSetProcessDEPPolicy)(dwFlags);
-			}
+			)
+		{
+			// VS2003:	Enable DEP (with ATL-thunk emulation) if not already set by system policy
+			//			or if the policy is not yet permanent.
+			//
+			// VS2005:	Enable DEP (without ATL-thunk emulation) if not already set by system policy
+			//			or linker "/NXCOMPAT" option or if the policy is not yet permanent. We should
+			//			not reach this code path at all because the "/NXCOMPAT" option is specified.
+			//			However, the code path is here for safety reasons.
+			dwFlags = PROCESS_DEP_ENABLE;
+			// VS2005: Disable ATL thunks.
+			dwFlags |= PROCESS_DEP_DISABLE_ATL_THUNK_EMULATION;
+			::SetProcessDEPPolicy(dwFlags);
 		}
 	}
 }
