diff --git a/srchybrid/OtherFunctions.cpp b/srchybrid/OtherFunctions.cpp
index ded9483..f818323 100644
--- a/srchybrid/OtherFunctions.cpp
+++ b/srchybrid/OtherFunctions.cpp
@@ -942,31 +942,27 @@ bool SelectDir(HWND hWnd, LPTSTR pszPath, LPCTSTR pszTitle, LPCTSTR pszDlgTitle)
 {
 	ASSERT(pszPath != NULL);
 	(void)::CoInitialize(NULL);
-	LPMALLOC pShlMalloc;
-	bool bResult = (SHGetMalloc(&pShlMalloc) == NOERROR);
-	if (bResult) {
-		BROWSEINFO BrsInfo = {};
-		BrsInfo.hwndOwner = hWnd;
-		BrsInfo.lpszTitle = (pszTitle != NULL) ? pszTitle : pszDlgTitle;
-		BrsInfo.ulFlags = BIF_VALIDATE | BIF_NEWDIALOGSTYLE | BIF_RETURNONLYFSDIRS | BIF_SHAREABLE | BIF_DONTGOBELOWDOMAIN;
-
-		BROWSEINIT BrsInit = {};
-		if (pszTitle != NULL || pszDlgTitle != NULL) {
-			// Need the 'BrowseCallbackProc' to set those strings
-			BrsInfo.lpfn = BrowseCallbackProc;
-			BrsInfo.lParam = (LPARAM)&BrsInit;
-			BrsInit.pszDlgTitle = (pszDlgTitle != NULL) ? pszDlgTitle : NULL/*pszTitle*/;
-			BrsInit.pszInitialDir = pszPath;
-		}
-
-		LPITEMIDLIST pidlBrowse;
-		if ((pidlBrowse = ::SHBrowseForFolder(&BrsInfo)) != NULL) {
-			bResult = ::SHGetPathFromIDList(pidlBrowse, pszPath);
-			pShlMalloc->Free(pidlBrowse);
-		}
-		pShlMalloc->Release();
-		::PathAddBackslash(pszPath);
-	}
+	bool bResult = true;
+	BROWSEINFO BrsInfo = {};
+	BrsInfo.hwndOwner = hWnd;
+	BrsInfo.lpszTitle = (pszTitle != NULL) ? pszTitle : pszDlgTitle;
+	BrsInfo.ulFlags = BIF_VALIDATE | BIF_NEWDIALOGSTYLE | BIF_RETURNONLYFSDIRS | BIF_SHAREABLE | BIF_DONTGOBELOWDOMAIN;
+
+	BROWSEINIT BrsInit = {};
+	if (pszTitle != NULL || pszDlgTitle != NULL) {
+		// Need the 'BrowseCallbackProc' to set those strings
+		BrsInfo.lpfn = BrowseCallbackProc;
+		BrsInfo.lParam = (LPARAM)&BrsInit;
+		BrsInit.pszDlgTitle = (pszDlgTitle != NULL) ? pszDlgTitle : NULL/*pszTitle*/;
+		BrsInit.pszInitialDir = pszPath;
+	}
+
+	LPITEMIDLIST pidlBrowse;
+	if ((pidlBrowse = ::SHBrowseForFolder(&BrsInfo)) != NULL) {
+		bResult = ::SHGetPathFromIDList(pidlBrowse, pszPath);
+		::CoTaskMemFree(pidlBrowse);
+	}
+	::PathAddBackslash(pszPath);
 	::CoUninitialize();
 	return bResult;
 }
diff --git a/srchybrid/PartFileConvert.cpp b/srchybrid/PartFileConvert.cpp
index d184930..ba3cbbc 100644
--- a/srchybrid/PartFileConvert.cpp
+++ b/srchybrid/PartFileConvert.cpp
@@ -608,49 +608,45 @@ void CPartFileConvertDlg::PostNcDestroy()
 void CPartFileConvertDlg::OnAddFolder()
 {
 	// browse...
-	LPMALLOC pMalloc = NULL;
-	if (SHGetMalloc(&pMalloc) == NOERROR) {
-		// buffer - a place to hold the file system pathname
-		TCHAR buffer[MAX_PATH];
-
-		// This struct holds the various options for the dialog
-		BROWSEINFO bi;
-		bi.hwndOwner = m_hWnd;
-		bi.pidlRoot = NULL;
-		bi.pszDisplayName = buffer;
-		CString title(GetResString(IDS_IMP_SELFOLDER));
-		bi.lpszTitle = title.GetBuffer(title.GetLength());
-		bi.ulFlags = BIF_EDITBOX | BIF_NEWDIALOGSTYLE | BIF_NONEWFOLDERBUTTON | BIF_SHAREABLE;
-		bi.lpfn = NULL;
-
-		// Now cause the dialog to appear.
-		LPITEMIDLIST pidlRoot;
-		if ((pidlRoot = SHBrowseForFolder(&bi)) != NULL) {
-			int reply;
-			if (thePrefs.IsExtControlsEnabled())
-				reply = LocMessageBox(IDS_IMP_DELSRC, MB_YESNOCANCEL | MB_DEFBUTTON2, 0);
-			else
-				reply = IDNO;
-
-			if (reply != IDCANCEL) {
-				bool removesrc = (reply == IDYES);
-
-				//
-				// Again, almost undocumented. How to get an ASCII pathname
-				// from the LPITEMIDLIST struct.
-				// I guess you just have to "know" this stuff.
-				//
-				if (SHGetPathFromIDList(pidlRoot, buffer)) {
-					// Do something with the converted string.
-					CPartFileConvert::ScanFolderToAdd(CString(buffer), removesrc);
-				}
+	// buffer - a place to hold the file system pathname
+	TCHAR buffer[MAX_PATH];
+
+	// This struct holds the various options for the dialog
+	BROWSEINFO bi;
+	bi.hwndOwner = m_hWnd;
+	bi.pidlRoot = NULL;
+	bi.pszDisplayName = buffer;
+	CString title(GetResString(IDS_IMP_SELFOLDER));
+	bi.lpszTitle = title.GetBuffer(title.GetLength());
+	bi.ulFlags = BIF_EDITBOX | BIF_NEWDIALOGSTYLE | BIF_NONEWFOLDERBUTTON | BIF_SHAREABLE;
+	bi.lpfn = NULL;
+
+	// Now cause the dialog to appear.
+	LPITEMIDLIST pidlRoot;
+	if ((pidlRoot = SHBrowseForFolder(&bi)) != NULL) {
+		int reply;
+		if (thePrefs.IsExtControlsEnabled())
+			reply = LocMessageBox(IDS_IMP_DELSRC, MB_YESNOCANCEL | MB_DEFBUTTON2, 0);
+		else
+			reply = IDNO;
+
+		if (reply != IDCANCEL) {
+			bool removesrc = (reply == IDYES);
+
+			//
+			// Again, almost undocumented. How to get an ASCII pathname
+			// from the LPITEMIDLIST struct.
+			// I guess you just have to "know" this stuff.
+			//
+			if (SHGetPathFromIDList(pidlRoot, buffer)) {
+				// Do something with the converted string.
+				CPartFileConvert::ScanFolderToAdd(CString(buffer), removesrc);
 			}
-
-			// Free the returned item identifier list using the
-			// shell's task allocator!Arghhhh.
-			pMalloc->Free(pidlRoot);
 		}
-		pMalloc->Release();
+
+		// Free the returned item identifier list using the
+		// shell's task allocator!Arghhhh.
+		::CoTaskMemFree(pidlRoot);
 	}
 }
 
diff --git a/srchybrid/TreeOptionsCtrl.cpp b/srchybrid/TreeOptionsCtrl.cpp
index 2a7ac27..ecfd488 100644
--- a/srchybrid/TreeOptionsCtrl.cpp
+++ b/srchybrid/TreeOptionsCtrl.cpp
@@ -2246,11 +2246,7 @@ void CTreeOptionsEdit::BrowseForFolder(const CString &sInitialFolder)
 
 		//avoid memory leaks by deleting the PIDL
 		//using the shells task allocator
-		IMalloc *pMalloc;
-		if (SUCCEEDED(SHGetMalloc(&pMalloc))) {
-			pMalloc->Free(pItemIDList);
-			pMalloc->Release();
-		}
+		::CoTaskMemFree(pItemIDList);
 	}
 }
 
