diff --git a/srchybrid/ClientListCtrl.cpp b/srchybrid/ClientListCtrl.cpp
index 6943e62..6620292 100644
--- a/srchybrid/ClientListCtrl.cpp
+++ b/srchybrid/ClientListCtrl.cpp
@@ -148,6 +148,7 @@ void CClientListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					m_pImageList->Draw(dc, iImage, point, ILD_NORMAL | INDEXTOOVERLAYMASK(uOverlayImage));
 					rcItem.left += 16 + sm_iLabelOffset - sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				dc.DrawText(sItem, &rcItem, MLC_DT_TEXT | uDrawTextAlignment);
diff --git a/srchybrid/DownloadClient.cpp b/srchybrid/DownloadClient.cpp
index 75959b5..e278bb5 100644
--- a/srchybrid/DownloadClient.cpp
+++ b/srchybrid/DownloadClient.cpp
@@ -2186,6 +2186,7 @@ void CUpDownClient::ProcessAICHFileHash(CSafeMemFile *data, CPartFile *file, con
 					pPartFile->SetAICHHashSetNeeded(true);
 					m_fHashsetRequestingAICH = false;
 				}
+				[[fallthrough]];
 			case DS_CONNECTED:
 			case DS_ONQUEUE:
 			case DS_NONEEDEDPARTS:
diff --git a/srchybrid/DownloadClientsCtrl.cpp b/srchybrid/DownloadClientsCtrl.cpp
index 5675d44..a174171 100644
--- a/srchybrid/DownloadClientsCtrl.cpp
+++ b/srchybrid/DownloadClientsCtrl.cpp
@@ -140,6 +140,7 @@ void CDownloadClientsCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					m_pImageList->Draw(dc, iImage, point, ILD_NORMAL | INDEXTOOVERLAYMASK(uOverlayImage));
 					rcItem.left += 16 + sm_iLabelOffset - sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				rcItem.right -= sm_iSubItemInset;
diff --git a/srchybrid/DownloadListCtrl.cpp b/srchybrid/DownloadListCtrl.cpp
index 2849fad..8084367 100644
--- a/srchybrid/DownloadListCtrl.cpp
+++ b/srchybrid/DownloadListCtrl.cpp
@@ -717,6 +717,7 @@ void CDownloadListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 			rcItem.left += iTreeOffset;
 			tree_end = rcItem.left;
 			rcItem.right -= iTreeOffset - sm_iLabelOffset;
+			[[fallthrough]];
 		default:
 			rcItem.left += sm_iLabelOffset;
 			rcItem.right -= sm_iLabelOffset;
@@ -1290,6 +1291,7 @@ BOOL CDownloadListCtrl::OnCommand(WPARAM wParam, LPARAM)
 							default:
 								if (partfile->GetCategory())
 									theApp.downloadqueue->StartNextFileIfPrefs(partfile->GetCategory());
+									[[fallthrough]];
 							case PS_PAUSED:
 								partfile->DeletePartFile();
 							}
diff --git a/srchybrid/DownloadQueue.cpp b/srchybrid/DownloadQueue.cpp
index c224761..9f7c445 100644
--- a/srchybrid/DownloadQueue.cpp
+++ b/srchybrid/DownloadQueue.cpp
@@ -1558,6 +1558,7 @@ void CDownloadQueue::KademliaSearchFile(uint32 nSearchID, const Kademlia::CUInt1
 			pcontactID->ToByteArray(cID);
 			ctemp->SetUserHash(cID);
 		}
+		[[fallthrough]];
 	case 2:
 		//Don't use this type... Some clients will process it wrong.
 		break;
diff --git a/srchybrid/EMSocket.cpp b/srchybrid/EMSocket.cpp
index 975aec0..c3c5a53 100644
--- a/srchybrid/EMSocket.cpp
+++ b/srchybrid/EMSocket.cpp
@@ -976,6 +976,7 @@ int CEMSocket::OnLayerCallback(std::vector<t_callbackMsg> &callbacks)
 				case PROXYERROR_NOCONN:
 					// We failed to connect to the proxy.
 					m_bProxyConnectFailed = true;
+					[[fallthrough]];
 				case PROXYERROR_REQUESTFAILED:
 					// We are connected to the proxy but it failed to connect to the peer.
 					if (thePrefs.GetVerbose() && iter->str && iter->str[0] != '\0')
diff --git a/srchybrid/EmuleDlg.cpp b/srchybrid/EmuleDlg.cpp
index 8d7bdec..ac22135 100644
--- a/srchybrid/EmuleDlg.cpp
+++ b/srchybrid/EmuleDlg.cpp
@@ -681,6 +681,7 @@ void CALLBACK CemuleDlg::StartupTimer(HWND /*hwnd*/, UINT /*uiMsg*/, UINT_PTR /*
 			++theApp.emuledlg->status;
 			theApp.sharedfiles->SetOutputCtrl(&theApp.emuledlg->sharedfileswnd->sharedfilesctrl);
 			++theApp.emuledlg->status;
+			[[fallthrough]];
 		case 1:
 			break;
 		case 2:
@@ -692,6 +693,7 @@ void CALLBACK CemuleDlg::StartupTimer(HWND /*hwnd*/, UINT /*uiMsg*/, UINT_PTR /*
 				LogError(LOG_STATUSBAR, _T("Failed to initialize server list - Unknown exception"));
 			}
 			++theApp.emuledlg->status;
+			[[fallthrough]];
 		case 3:
 			break;
 		case 4:
@@ -1302,6 +1304,7 @@ void CemuleDlg::ProcessED2KLink(LPCWSTR pszData)
 				SetActiveDialog(searchwnd);
 				searchwnd->ProcessEd2kSearchLinkRequest(pListLink->GetSearchTerm());
 			}
+			[[fallthrough]];
 		default:
 			break;
 		}
diff --git a/srchybrid/FileInfoDialog.cpp b/srchybrid/FileInfoDialog.cpp
index b9746b6..0332d0a 100644
--- a/srchybrid/FileInfoDialog.cpp
+++ b/srchybrid/FileInfoDialog.cpp
@@ -927,6 +927,7 @@ bool CGetMediaInfoThread::GetMediaInfo(HWND hWndOwner, const CShareableFile *pFi
 				case MPEGVERSION_1:
 					mi->strAudioFormat = _T("MPEG-1,");
 					mi->audio.wFormatTag = WAVE_FORMAT_MPEGLAYER3;
+					[[fallthrough]];
 				default:
 					break;
 				}
@@ -941,6 +942,7 @@ bool CGetMediaInfoThread::GetMediaInfo(HWND hWndOwner, const CShareableFile *pFi
 					break;
 				case MPEGLAYER_I:
 					mi->strAudioFormat += _T("Layer 1");
+					[[fallthrough]];
 				default:
 					break;
 				}
diff --git a/srchybrid/IrcMain.cpp b/srchybrid/IrcMain.cpp
index 36c8634..8efab03 100644
--- a/srchybrid/IrcMain.cpp
+++ b/srchybrid/IrcMain.cpp
@@ -500,7 +500,7 @@ void CIrcMain::ParseMessage(const CString &sRawMessage)
 					m_pIRCSocket->SendString(strCommand);
 				}
 				ParsePerform();
-				/* fall through */
+				[[fallthrough]];
 			case 2:
 			case 3:
 			case 4:
diff --git a/srchybrid/IrcWnd.cpp b/srchybrid/IrcWnd.cpp
index 13daeab..773423c 100644
--- a/srchybrid/IrcWnd.cpp
+++ b/srchybrid/IrcWnd.cpp
@@ -435,6 +435,7 @@ BOOL CIrcWnd::OnCommand(WPARAM wParam, LPARAM)
 		break;
 	case IDC_SMILEY:
 		OnBnClickedSmiley();
+		[[fallthrough]];
 	default:
 		return TRUE;
 	}
diff --git a/srchybrid/ListCtrlEditable.cpp b/srchybrid/ListCtrlEditable.cpp
index 584f2e6..a0bb28e 100644
--- a/srchybrid/ListCtrlEditable.cpp
+++ b/srchybrid/ListCtrlEditable.cpp
@@ -250,6 +250,7 @@ void CEditableListCtrl::OnLButtonDown(UINT nFlags, CPoint point)
 	switch (m_iCol) {
 	case 0:
 		m_iCol = 1; //move to column 1
+		[[fallthrough]];
 	case 1:
 		ShowEdit(); //start editing
 	}
@@ -261,6 +262,7 @@ void CEditableListCtrl::OnVScroll(UINT nSBCode, UINT nPos, CScrollBar *pScrollBa
 		switch (nSBCode) {
 		case SB_ENDSCROLL:
 			ShowEdit();
+			[[fallthrough]];
 		case SB_LINEDOWN:
 		case SB_LINEUP:
 			break;
diff --git a/srchybrid/PartFile.cpp b/srchybrid/PartFile.cpp
index 8f45164..7d277b0 100644
--- a/srchybrid/PartFile.cpp
+++ b/srchybrid/PartFile.cpp
@@ -2245,6 +2245,7 @@ uint32 CPartFile::Process(uint32 factor, UINT icounter)
 			case DS_DOWNLOADING:
 				ASSERT(cur_src->socket);
 				UpdateDownloadLimit(cur_src, factor);
+				[[fallthrough]];
 			case DS_BANNED: // Do nothing with this client.
 				break;
 			case DS_LOWTOLOWIP: // Check if something has changed with our or their ID state.
@@ -2294,6 +2295,7 @@ uint32 CPartFile::Process(uint32 factor, UINT icounter)
 				if (theApp.IsConnected() && cur_src->GetTimeUntilReask() < MIN2MS(2) && cur_src->GetTimeUntilReask() > SEC2MS(1) && curTick >= cur_src->GetLastTriedToConnectTime() + MIN2MS(20)) // ZZ:DownloadManager (one re-ask timestamp for each file)
 					cur_src->UDPReaskForDownload();
 
+					[[fallthrough]];
 			case DS_CONNECTING:
 			case DS_TOOMANYCONNS:
 			case DS_TOOMANYCONNSKAD:
@@ -4095,6 +4097,7 @@ void CPartFile::FlushBuffer(bool bForceICH, bool bNoAICH)
 			switch (item->flushed) {
 			case PB_READY:
 				ASSERT(!pThread || !pThread->IsRunning());
+				[[fallthrough]];
 			case PB_PENDING:
 				continue;
 			case PB_ERROR:
diff --git a/srchybrid/Preferences.cpp b/srchybrid/Preferences.cpp
index 8508c78..1bf3a46 100644
--- a/srchybrid/Preferences.cpp
+++ b/srchybrid/Preferences.cpp
@@ -2954,6 +2954,7 @@ CString CPreferences::GetDefaultDirectory(EDefaultDirectory eDirectory, bool bCr
 								strSelectedDataBaseDirectory = strPublicDownloads + _T("eMule\\");
 								strSelectedConfigBaseDirectory = strProgramData + _T("eMule\\");
 								strSelectedExpansionBaseDirectory = strProgramData + _T("eMule\\");
+								[[fallthrough]];
 							case 2: //program directory
 								break;
 							default:
diff --git a/srchybrid/QueueListCtrl.cpp b/srchybrid/QueueListCtrl.cpp
index 2825b14..1ee31ac 100644
--- a/srchybrid/QueueListCtrl.cpp
+++ b/srchybrid/QueueListCtrl.cpp
@@ -156,6 +156,7 @@ void CQueueListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					m_pImageList->Draw(dc, iImage, point, ILD_NORMAL | INDEXTOOVERLAYMASK(uOverlayImage));
 					rcItem.left += 16 + sm_iLabelOffset - sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				rcItem.right -= sm_iSubItemInset;
diff --git a/srchybrid/SearchListCtrl.cpp b/srchybrid/SearchListCtrl.cpp
index 4fa5238..30a17c5 100644
--- a/srchybrid/SearchListCtrl.cpp
+++ b/srchybrid/SearchListCtrl.cpp
@@ -1291,6 +1291,7 @@ void CSearchListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 			tree_start = rcItem.left + 1;
 			rcItem.left += min(8, iColumnWidth);
 			tree_end = rcItem.left;
+			[[fallthrough]];
 		default:
 			rcItem.left += sm_iLabelOffset;
 			if (rcItem.left < rcItem.right && HaveIntersection(rcClient, rcItem))
@@ -1417,8 +1418,10 @@ void CSearchListCtrl::DrawSourceChild(CDC &dc, int nColumn, LPRECT lpRect, UINT
 		{
 			lpRect->left += 16;
 		}
+		[[fallthrough]];
 	default:
 		dc.DrawText(sItem, lpRect, MLC_DT_TEXT | uDrawTextAlignment);
+		[[fallthrough]];
 	case 4: // file type
 	case 5: // file hash
 		break;
@@ -1436,6 +1439,7 @@ void CSearchListCtrl::DrawSourceParent(CDC &dc, int nColumn, LPRECT lpRect, UINT
 		{
 			lpRect->left += 16;
 		}
+		[[fallthrough]];
 	default:
 		dc.DrawText(sItem, lpRect, MLC_DT_TEXT | uDrawTextAlignment);
 		break;
diff --git a/srchybrid/ServerListCtrl.cpp b/srchybrid/ServerListCtrl.cpp
index 84a5c0c..6f618cd 100644
--- a/srchybrid/ServerListCtrl.cpp
+++ b/srchybrid/ServerListCtrl.cpp
@@ -236,6 +236,7 @@ void CServerListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					m_pImageList->Draw(dc, iImage, point, ILD_NORMAL | INDEXTOOVERLAYMASK(uOverlayImage));
 					rcItem.left += 16 + sm_iLabelOffset - sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				dc.DrawText(sItem, &rcItem, MLC_DT_TEXT | uDrawTextAlignment);
diff --git a/srchybrid/SharedDirsTreeCtrl.cpp b/srchybrid/SharedDirsTreeCtrl.cpp
index c6fc64c..7e57bc6 100644
--- a/srchybrid/SharedDirsTreeCtrl.cpp
+++ b/srchybrid/SharedDirsTreeCtrl.cpp
@@ -668,6 +668,7 @@ BOOL CSharedDirsTreeCtrl::OnCommand(WPARAM wParam, LPARAM)
 						break;
 					case MP_PRIOVERYHIGH:
 						pri = PR_VERYHIGH;
+						[[fallthrough]];
 					case MP_PRIOAUTO:
 						break;
 					default:
@@ -682,6 +683,7 @@ BOOL CSharedDirsTreeCtrl::OnCommand(WPARAM wParam, LPARAM)
 					}
 				}
 			}
+			[[fallthrough]];
 		default:
 			break;
 		}
diff --git a/srchybrid/SharedFilesCtrl.cpp b/srchybrid/SharedFilesCtrl.cpp
index e9a69b1..75f82da 100644
--- a/srchybrid/SharedFilesCtrl.cpp
+++ b/srchybrid/SharedFilesCtrl.cpp
@@ -565,6 +565,7 @@ void CSharedFilesCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					}
 					rcItem.left -= sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				rcItem.right -= sm_iSubItemInset;
diff --git a/srchybrid/TaskbarNotifier.cpp b/srchybrid/TaskbarNotifier.cpp
index eacd888..33ae4d2 100644
--- a/srchybrid/TaskbarNotifier.cpp
+++ b/srchybrid/TaskbarNotifier.cpp
@@ -450,6 +450,7 @@ void CTaskbarNotifier::Show(LPCTSTR pszCaption, TbnMsg nMsgType, LPCTSTR pszLink
 		}
 		ShowWindow(SW_SHOWNOACTIVATE);
 		SetTimer(IDT_APPEARING, m_dwShowEvents, NULL);
+		[[fallthrough]];
 	case IDT_APPEARING:
 	case IDT_WAITING:
 		RedrawWindow(&m_rcText);
diff --git a/srchybrid/TransferWnd.cpp b/srchybrid/TransferWnd.cpp
index 5182375..4f75c5d 100644
--- a/srchybrid/TransferWnd.cpp
+++ b/srchybrid/TransferWnd.cpp
@@ -453,6 +453,7 @@ void CTransferWnd::UpdateListCount(EWnd2 listindex, int iCount /*=-1*/)
 			strBuffer.Format(_T("%s (%i)"), (LPCTSTR)GetResString(IDS_DOWNLOADING), (iCount < 0) ? downloadclientsctrl.GetItemCount() : iCount);
 			m_btnWnd1.SetWindowText(strBuffer);
 		}
+		[[fallthrough]];
 	default:
 		/*ASSERT(0)*/;
 	}
diff --git a/srchybrid/UPnPImplWinServ.cpp b/srchybrid/UPnPImplWinServ.cpp
index 91c5bf6..0ca6b11 100644
--- a/srchybrid/UPnPImplWinServ.cpp
+++ b/srchybrid/UPnPImplWinServ.cpp
@@ -830,6 +830,7 @@ HRESULT CUPnPImplWinServ::InvokeAction(ServicePointer pService
 			// If connection services return value is empty
 			// then OUT arguments are returned
 			GetStringFromOutArgs(&vaOutArgs, strResult);
+			[[fallthrough]];
 		default:
 			bInvalid = true;
 		}
diff --git a/srchybrid/UploadListCtrl.cpp b/srchybrid/UploadListCtrl.cpp
index dee6c48..1dfbbe6 100644
--- a/srchybrid/UploadListCtrl.cpp
+++ b/srchybrid/UploadListCtrl.cpp
@@ -153,6 +153,7 @@ void CUploadListCtrl::DrawItem(LPDRAWITEMSTRUCT lpDrawItemStruct)
 					m_pImageList->Draw(dc, iImage, point, ILD_NORMAL | INDEXTOOVERLAYMASK(uOverlayImage));
 					rcItem.left += 16 + sm_iLabelOffset - sm_iSubItemInset;
 				}
+				[[fallthrough]];
 			default: //any text column
 				rcItem.left += sm_iSubItemInset;
 				rcItem.right -= sm_iSubItemInset;
diff --git a/srchybrid/kademlia/kademlia/Search.cpp b/srchybrid/kademlia/kademlia/Search.cpp
index 4a3ec99..7e67042 100644
--- a/srchybrid/kademlia/kademlia/Search.cpp
+++ b/srchybrid/kademlia/kademlia/Search.cpp
@@ -106,6 +106,7 @@ CSearch::~CSearch()
 	case FINDSOURCE: // maybe also exclude
 		if (m_uClosestDistantFound != 0)
 			CKademlia::StatsAddClosestDistance(m_uClosestDistantFound);
+			[[fallthrough]];
 	default: // NODE, NODESPECIAL, NODEFWCHECKUDP, FINDBUDDY
 		break;
 	}
diff --git a/srchybrid/kademlia/net/PacketTracking.cpp b/srchybrid/kademlia/net/PacketTracking.cpp
index a11a558..e0dc86b 100644
--- a/srchybrid/kademlia/net/PacketTracking.cpp
+++ b/srchybrid/kademlia/net/PacketTracking.cpp
@@ -137,6 +137,7 @@ int CPacketTracking::InTrackListIsAllowedPacket(uint32 uIP, uint8 byOpcode, bool
 		break;
 	case KADEMLIA_FIREWALLED2_REQ:
 		byOpcode = KADEMLIA_FIREWALLED_REQ;
+		[[fallthrough]];
 	case KADEMLIA_FIREWALLED_REQ:
 	case KADEMLIA_FINDBUDDY_REQ:
 		token = MIN2MS(1) / 2;
