diff --git a/srchybrid/DownloadListCtrl.cpp b/srchybrid/DownloadListCtrl.cpp
index 48231c9..97e935c 100644
--- a/srchybrid/DownloadListCtrl.cpp
+++ b/srchybrid/DownloadListCtrl.cpp
@@ -869,8 +869,6 @@ void CDownloadListCtrl::OnLvnItemActivate(LPNMHDR pNMHDR, LRESULT *pResult)
 	*pResult = 0;
 }
 
-#pragma warning(push)
-#pragma warning(disable:4302 4311)
 void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 {
 	int iSel = GetNextItem(-1, LVIS_SELECTED);
@@ -944,7 +942,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 					uPrioMenuItem = 0;
 			}
 
-			m_FileMenu.EnableMenuItem((UINT)m_PrioMenu.m_hMenu, iFilesNotDone > 0 ? MF_ENABLED : MF_GRAYED);
+			m_FileMenu.EnableMenuItem(HandleToULong(m_PrioMenu.m_hMenu), iFilesNotDone > 0 ? MF_ENABLED : MF_GRAYED);
 			m_PrioMenu.CheckMenuRadioItem(MP_PRIOLOW, MP_PRIOAUTO, uPrioMenuItem, 0);
 
 			// enable commands if there is at least one item which can be used for the action
@@ -967,7 +965,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 				m_PreviewMenu.EnableMenuItem(MP_PREVIEW, (iSelectedItems == 1 && iFilesToPreview == 1) ? MF_ENABLED : MF_GRAYED);
 				m_PreviewMenu.EnableMenuItem(MP_PAUSEONPREVIEW, iFilesCanPauseOnPreview > 0 ? MF_ENABLED : MF_GRAYED);
 				m_PreviewMenu.CheckMenuItem(MP_PAUSEONPREVIEW, (iSelectedItems > 0 && iFilesDoPauseOnPreview == iSelectedItems) ? MF_CHECKED : MF_UNCHECKED);
-				m_FileMenu.EnableMenuItem((UINT)m_PreviewMenu.m_hMenu, m_PreviewMenu.HasEnabledItems() ? MF_ENABLED : MF_GRAYED);
+				m_FileMenu.EnableMenuItem(HandleToULong(m_PreviewMenu.m_hMenu), m_PreviewMenu.HasEnabledItems() ? MF_ENABLED : MF_GRAYED);
 
 				if (iPreviewMenuEntries > 0)
 					if (thePrefs.GetExtraPreviewWithMenu())
@@ -997,7 +995,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 			int total;
 			m_FileMenu.EnableMenuItem(MP_CLEARCOMPLETED, GetCompleteDownloads(m_curTab, total) > 0 ? MF_ENABLED : MF_GRAYED);
 			if (thePrefs.IsExtControlsEnabled()) {
-				m_FileMenu.EnableMenuItem((UINT)m_SourcesMenu.m_hMenu, MF_ENABLED);
+				m_FileMenu.EnableMenuItem(HandleToULong(m_SourcesMenu.m_hMenu), MF_ENABLED);
 				m_SourcesMenu.EnableMenuItem(MP_ADDSOURCE, (iSelectedItems == 1 && iFilesToStop == 1) ? MF_ENABLED : MF_GRAYED);
 				m_SourcesMenu.EnableMenuItem(MP_SETSOURCELIMIT, (iFilesNotDone == iSelectedItems) ? MF_ENABLED : MF_GRAYED);
 			}
@@ -1036,9 +1034,9 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 			VERIFY(m_FileMenu.RemoveMenu(m_FileMenu.GetMenuItemCount() - 1, MF_BYPOSITION));
 			if (iPreviewMenuEntries)
 				if (thePrefs.IsExtControlsEnabled() && !thePrefs.GetExtraPreviewWithMenu())
-					VERIFY(m_PreviewMenu.RemoveMenu((UINT)PreviewWithMenu.m_hMenu, MF_BYCOMMAND));
+					VERIFY(m_PreviewMenu.RemoveMenu(HandleToULong(PreviewWithMenu.m_hMenu), MF_BYCOMMAND));
 				else
-					VERIFY(m_FileMenu.RemoveMenu((UINT)PreviewWithMenu.m_hMenu, MF_BYCOMMAND));
+					VERIFY(m_FileMenu.RemoveMenu(HandleToULong(PreviewWithMenu.m_hMenu), MF_BYCOMMAND));
 
 			VERIFY(WebMenu.DestroyMenu());
 			VERIFY(CatsMenu.DestroyMenu());
@@ -1079,7 +1077,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 		}
 	} else { // nothing selected
 		int total;
-		m_FileMenu.EnableMenuItem((UINT)m_PrioMenu.m_hMenu, MF_GRAYED);
+		m_FileMenu.EnableMenuItem(HandleToULong(m_PrioMenu.m_hMenu), MF_GRAYED);
 		m_FileMenu.EnableMenuItem(MP_CANCEL, MF_GRAYED);
 		m_FileMenu.EnableMenuItem(MP_PAUSE, MF_GRAYED);
 		m_FileMenu.EnableMenuItem(MP_STOP, MF_GRAYED);
@@ -1087,7 +1085,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 		m_FileMenu.EnableMenuItem(MP_OPEN, MF_GRAYED);
 
 		if (thePrefs.IsExtControlsEnabled()) {
-			m_FileMenu.EnableMenuItem((UINT)m_PreviewMenu.m_hMenu, MF_GRAYED);
+			m_FileMenu.EnableMenuItem(HandleToULong(m_PreviewMenu.m_hMenu), MF_GRAYED);
 			if (!thePrefs.GetPreviewPrio()) {
 				m_PreviewMenu.EnableMenuItem(MP_TRY_TO_GET_PREVIEW_PARTS, MF_GRAYED);
 				m_PreviewMenu.CheckMenuItem(MP_TRY_TO_GET_PREVIEW_PARTS, MF_UNCHECKED);
@@ -1107,7 +1105,7 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 		m_FileMenu.EnableMenuItem(MP_PASTE, theApp.IsEd2kFileLinkInClipboard() ? MF_ENABLED : MF_GRAYED);
 		m_FileMenu.SetDefaultItem(UINT_MAX);
 		if (m_SourcesMenu)
-			m_FileMenu.EnableMenuItem((UINT)m_SourcesMenu.m_hMenu, MF_GRAYED);
+			m_FileMenu.EnableMenuItem(HandleToULong(m_SourcesMenu.m_hMenu), MF_GRAYED);
 		m_FileMenu.EnableMenuItem(MP_SEARCHRELATED, MF_GRAYED);
 		m_FileMenu.EnableMenuItem(MP_FIND, GetItemCount() > 0 ? MF_ENABLED : MF_GRAYED);
 
@@ -1135,7 +1133,6 @@ void CDownloadListCtrl::OnContextMenu(CWnd*, CPoint point)
 		VERIFY(WebMenu.DestroyMenu());
 	}
 }
-#pragma warning(pop)
 
 void CDownloadListCtrl::FillCatsMenu(CMenu &rCatsMenu, int iFilesInCats)
 {
diff --git a/srchybrid/kademlia/kademlia/Search.cpp b/srchybrid/kademlia/kademlia/Search.cpp
index 7e67042..3904c51 100644
--- a/srchybrid/kademlia/kademlia/Search.cpp
+++ b/srchybrid/kademlia/kademlia/Search.cpp
@@ -1053,8 +1053,6 @@ void CSearch::ProcessResultNotes(const CUInt128 &uAnswer, TagList &rlistInfo)
 	}
 }
 
-#pragma warning(push)
-#pragma warning(disable:4312)
 void CSearch::ProcessResultKeyword(const CUInt128 &uAnswer, TagList &rlistInfo, uint32 uFromIP, uint16 uFromPort)
 {
 	// Find the contact who sent the answer - we need to know its protocol version
@@ -1226,12 +1224,11 @@ void CSearch::ProcessResultKeyword(const CUInt128 &uAnswer, TagList &rlistInfo,
 		, TAGTYPE_STRING, TAG_MEDIA_ARTIST,		(LPCWSTR)sArtist
 		, TAGTYPE_STRING, TAG_MEDIA_ALBUM,		(LPCWSTR)sAlbum
 		, TAGTYPE_STRING, TAG_MEDIA_TITLE,		(LPCWSTR)sTitle
-		, TAGTYPE_UINT32, TAG_MEDIA_LENGTH,		(LPCTSTR)uLength		//uint32
-		, TAGTYPE_UINT32, TAG_MEDIA_BITRATE,	(LPCTSTR)uBitrate		//uint32
+		, TAGTYPE_UINT32, TAG_MEDIA_LENGTH,		(LPCTSTR)(UINT_PTR)uLength		//uint32
+		, TAGTYPE_UINT32, TAG_MEDIA_BITRATE,	(LPCTSTR)(UINT_PTR)uBitrate		//uint32
 		, TAGTYPE_STRING, TAG_MEDIA_CODEC,		(LPCWSTR)sCodec
-		, TAGTYPE_UINT32, TAG_SOURCES,			(LPCTSTR)uAvailability);//uint32
+		, TAGTYPE_UINT32, TAG_SOURCES,			(LPCTSTR)(UINT_PTR)uAvailability);//uint32
 }
-#pragma warning(pop)
 
 void CSearch::SendFindValue(CContact *pContact, bool bReAskMore)
 {
