diff --git a/srchybrid/lang/fr_BR.rc b/srchybrid/lang/fr_BR.rc
index 7140734..772ecbe 100644
--- a/srchybrid/lang/fr_BR.rc
+++ b/srchybrid/lang/fr_BR.rc
@@ -1,5 +1,6 @@
 ﻿//Microsoft Developer Studio generated resource script.
 //
+#pragma code_page(65001)
 #include "fr_BR.h"
 
 #define APSTUDIO_READONLY_SYMBOLS
@@ -261,8 +262,8 @@ BEGIN
 	IDS_DL_REMAINS	"Amzer o chom"
 	IDS_UPDATEFROM	"Nevesaat adalek al lec'hienn : (filter.dat/PeerGuardian)"
 	IDS_QL_USERNAME	"Anv implijour"
-	IDS_LOADURL	"Fazi ket keweriet e-pad keweriañ ar pakadoù (%s)"
-	IDS_DELETEAFTERALLOC	"Prizadur"
+	IDS_LOADURL	"Kargañ"
+//	IDS_DELETEAFTERALLOC	"File '%s' is currently allocating disk space and will be deleted afterwards"
 	IDS_CLIENTS 		"Ostizien"
 	IDS_QL_RATING	"O prizañ"
 	IDS_CONNECTINGTO	"O kennaskañ ouzh %s (%s:%u)..."
@@ -347,8 +348,8 @@ BEGIN
 	IDS_ERR_ALREADY_DOWNLOADED	"Pellgarget eo bet ganeoc'h dija ar restr : %s"
 	IDS_ERR_SERVERMET_BAD	"Faziet eo ar restr known.met, n'haller ket he c'hargañ"
 	IDS_ERR_SERVERMET_UNKNOWN	"Fazi lenn dianav gant known.met : %s, n'haller ket e gargañ"
-	IDS_ERR_CONNECTIONERROR	"Fazi e-pad ar c'hennaksañ ouzh %s (%s:%u) : %u"
-	IDS_ERR_SOCKET	"Fazi e-barzh serversocket : %s (%s:%u) : %u"
+	IDS_ERR_CONNECTIONERROR	"Fazi e-pad ar c'hennaksañ ouzh %s (%s:%u) : %s"
+	IDS_ERR_SOCKET	"Fazi e-barzh serversocket : %s (%s:%u) : %s"
 	IDS_SHAREDFOUND	"%u restr eskemmet anavezet zo bet kavet"
 	IDS_SHAREDFOUNDHASHING	"%u restr eskemmet anavezet zo bet kavet, hachage %u nouveaux fichiers"
 END
@@ -532,7 +533,7 @@ BEGIN
 	IDS_ERR_BADPORT	"Niverenn borzh direizh"
 	IDS_ERR_NOHELLO	"Goulenn a ra un dra bennak hep lârout Demat"
 	IDS_ERR_INVALIDPACKET 	"Direizh pe gwastet eo ar pakad resevet"
-	IDS_ERR_CLIENTERROR	"An ostiz '%s' (IP:%s) en deus degaset ur fazi gantañ: %u. Digennaskañ an ostiz !"
+	IDS_ERR_CLIENTERROR	"An ostiz '%s' (IP:%s) en deus degaset ur fazi gantañ: %s. Digennaskañ an ostiz !"
 END
 
 STRINGTABLE FIXED IMPURE
@@ -690,7 +691,7 @@ BEGIN
 	IDS_IRC_HASPARTED	"* %s a zo deuet er-maez eus %s (%s)"
 	IDS_IRC_WASKICKEDBY	"* %s a zo bet lakaet er-maez gant %s (%s)"
 	IDS_DOWNLOAD_ALPHABETICAL	"Pellgargañ diouzh an urzh lizherenneg"
-	IDS_IRC_HASQUIT 				"* %s a zo aet"
+	IDS_IRC_HASQUIT 				"* %s a zo aet (%s)"
 	IDS_IRC_NOWKNOWNAS	"* %s a zo graet %s anezhañ bremañ"
 	IDS_IRC_SETSMODE	"* %s en deus kemmet ar mod gantañ : %s %s"
 	IDS_IRC_NOTSUPPORTED	"** Modoù zo a zo bet kemmet ha n'int ket skoret. Ret eo deoc'h mont er-maez ha dont en-dro er saloñs a-benn nevesaat ho rollad moranvioù.."
@@ -815,7 +816,7 @@ BEGIN
 	IDS_PRIOAUTOHIGH	"Emgefre [Uhel]"
 	IDS_PRIOAUTORELEASE	"Emgefre [Krenn]"
 	IDS_PRIOAUTOLOW	"Emgefre [Izel]"
-	IDS_ERR_CONTIMEOUT	"Oc'h esaeañ kennaskañ ouzh ""%s"" : re hir"
+	IDS_ERR_CONTIMEOUT	"Oc'h esaeañ kennaskañ ouzh ""%s"" (%s:%u) : re hir"
 	IDS_OUTOFSERVERS	"N'eus servijer vak ken evit kennaskañ."
 	IDS_PW_CONLIMITS	"Harzoù kennaskañ "
 	IDS_MAXCON5SECLABEL	"Uhel. kennaskañ nevez / 5 eil"
@@ -1430,16 +1431,16 @@ BEGIN
 	IDS_KADDISTANCE	"Pellder"
 	IDS_KEY	"Alc'hwez"
 	IDS_NUMBER	"Niver"
-	IDS_KAD_SEARCHSRC	"Klask orinoù (%u:%u)"
-	IDS_KAD_SEARCHKW	"Klask gerioù-tremen (%u:%u)"
+	IDS_KAD_SEARCHSRC	"Klask orinoù"
+	IDS_KAD_SEARCHKW	"Klask gerioù-tremen"
 END
 
 STRINGTABLE FIXED IMPURE
 BEGIN
-	IDS_KAD_NODE	"Dizoleiñ un neudennad (%u:%u)"
-	IDS_KAD_STOREFILE	"Enrollañ restr (%u:%u)"
-	IDS_KAD_STOREKW	"Enrollañ ger alc'hwez (%u:%u)"
-	IDS_KAD_UNKNOWN	"Dianav (%u:%u)"
+	IDS_KAD_NODE	"Dizoleiñ un neudennad"
+	IDS_KAD_STOREFILE	"Enrollañ restr"
+	IDS_KAD_STOREKW	"Enrollañ ger alc'hwez"
+	IDS_KAD_UNKNOWN	"Dianav"
 	IDS_SRV_TCPCOMPR	"Gwaskerezh TCP"
 	IDS_KADSEARCHLAB	"Klaskoù red"
 	IDS_SRV_UDPSR	"Protokol UDP astennet evit ar rekedoù orin"
@@ -1501,7 +1502,7 @@ BEGIN
 	IDS_INSTALL_SKIN	"Staliañ ur gwiskadur eMule"
     		IDS_INSTALL_SKIN_NODIR	"Teuliad ebet a zo bet diuzet evit gwiskadurioù an etrefas eMule, pe n'eo ket haezus an teuliad-mañ c'hoazh!"
 	IDS_INSTALL_SKIN_DIR_ERROR	"Fazi o sevel an teuliad ""%s"" evit gwiskadur an etrefas eMule - %s."
-	IDS_INSTALL_SKIN_FILE_ERROR "Fazi oc'h eztennañ ar restr ""%s"" adalek gwiskadur an etrefas eMule diwar ar restr lec'hel ""%s""."
+	IDS_INSTALL_SKIN_FILE_ERROR "Fazi oc'h eztennañ ar restr ""%s"" adalek gwiskadur an etrefas eMule diwar ar restr lec'hel ""%s"" - %s"
 	IDS_INSTALL_SKIN_PKG_ERROR	"N'eo ket ar restr-mañ ur gwiskadur reizh evit an etrefas eMule."
 	IDS_NEXT_REASK	"Goulenn o tont gant an ostiz-mañ"
 	IDS_A4AF_FILES	"Restroù A4AF"
diff --git a/srchybrid/lang/nn_NO.rc b/srchybrid/lang/nn_NO.rc
index 974fbb5..38dc709 100644
--- a/srchybrid/lang/nn_NO.rc
+++ b/srchybrid/lang/nn_NO.rc
@@ -1,4 +1,5 @@
-﻿#include "nn_NO.h"
+﻿#pragma code_page(65001)
+#include "nn_NO.h"
 
 #define APSTUDIO_READONLY_SYMBOLS
 /////////////////////////////////////////////////////////////////////////////
diff --git a/srchybrid/lang/sl_SI.rc b/srchybrid/lang/sl_SI.rc
index 84a1688..8301017 100644
--- a/srchybrid/lang/sl_SI.rc
+++ b/srchybrid/lang/sl_SI.rc
@@ -561,9 +561,9 @@ BEGIN
     IDS_CD_RATING           "Ocena (skupna):"
     IDS_CD_USCORE           "Točke v vrsti:"
     IDS_CD_TITLE            "Podrobnosti o uporabniku"
-    IDS_EMULEW              "Kliknite tukaj,"
+    IDS_EMULEW              "Kliknite"
     IDS_EMULEW2             "da preverite, če je na voljo nova različica,"
-    IDS_EMULEW3             ""
+    IDS_EMULEW3             "tukaj,"
     IDS_PW_MAXSOURCES       "Maks. št. virov na datoteko"
     IDS_SHAREDFILESCOUNT    "Število deljenih datotek: %u"
     IDS_SF_SIZE             "Celotna velikost deljenih datotek: %s"
