diff --git a/srchybrid/lang/ar_AE.rc b/srchybrid/lang/ar_AE.rc
index 4e4a867..bb3f433 100644
--- a/srchybrid/lang/ar_AE.rc
+++ b/srchybrid/lang/ar_AE.rc
@@ -1971,6 +1971,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "طريقة التوثيق"
+    IDS_BOOTSTRAPFAILED     "فشل الاتصال بشبكة كاد. ملف nodes.dat قد يكون قديما أو هناك مشكلة في الاتصال."
+    IDS_CERT_CREATED        "تم إنشاء شهادة جديدة; الرقم التسلسلي %d"
+    IDS_CERT_ERR_CREATE     "لم يتم إنشاء الشهادة (انظر سجل Verbose)"
+    IDS_CERT_NOT_FOUND      "ملف الشهادة غير موجود"
+    IDS_CONTENTDB           "ContentDB (ويب)"
+    IDS_DELETEAFTERFILEOP   "الملف '%s' سيتم حذفه بعد انتهاء العملية الحالية"
+    IDS_DL_RESUMED          "استئناف تحميل ""%s"""
+    IDS_ED2KSERVER          "خادم eD2K"
+    IDS_ENABLEIMPORTPARTS   "فعل استيراد الأجزاء"
+    IDS_ERR_REGEXP          "تعبير نمطي غير صالح"
+    IDS_IMPORTPARTS         "استيراد الأجزاء..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "تم إلغاء الاستيراد: لا يمكن الاستيراد إلى ملف مكتمل."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "تم إلغاء الاستيراد: تعذر فتح %s."
+    IDS_IMPORTPARTS_IMPORTSTART "استيراد %i أجزاء إلى %s."
+    IDS_IMPORTPARTS_STOP    "أوقف استيراد الأجزاء"
+    IDS_IRC_BAN             "احظر"
+    IDS_IRC_ENABLEUTF8      "فعل UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "تجاهل رسائل Ping? Pong!"
+    IDS_ITALIC              "تعيين الخط المائل"
+    IDS_KEY_NOT_FOUND       "ملف المفتاح غير موجود"
+    IDS_PREVIEWREQUESTED    "تم طلب المعاينة - الرجاء الانتظار"
+    IDS_UNCADD              "اضف مسار UNC..."
+    IDS_WEB_CERT            "الشهادة"
+    IDS_WEB_GENERATE        "أنشئ شهادة جديدة"
+    IDS_WEB_HTTPS           "استخدم HTTPS"
 END
 
 #endif    // Arabic (IQ) resources
diff --git a/srchybrid/lang/ba_BA.rc b/srchybrid/lang/ba_BA.rc
index b1bbb58..556655a 100644
--- a/srchybrid/lang/ba_BA.rc
+++ b/srchybrid/lang/ba_BA.rc
@@ -1978,6 +1978,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Basque resources
diff --git a/srchybrid/lang/bg_BG.rc b/srchybrid/lang/bg_BG.rc
index 98090d7..7b26dbe 100644
--- a/srchybrid/lang/bg_BG.rc
+++ b/srchybrid/lang/bg_BG.rc
@@ -1962,6 +1962,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Включи ефектите на Windows 7 за лентата със задачи"
     IDS_LANMODE             "Режим LAN"
     IDS_OPENPREFINI         "Отвори 'preferences.ini'"
+    IDS_AUTHMETHOD          "Метод за удостоверяване"
+    IDS_BOOTSTRAPFAILED     "Неуспешно свързване с Kad. Вашият файл nodes.dat може да е остарял или има проблем с връзката."
+    IDS_CERT_CREATED        "Създаден е нов сертификат; сериен номер %d"
+    IDS_CERT_ERR_CREATE     "Сертификатът не е създаден (вижте дневника Verbose)"
+    IDS_CERT_NOT_FOUND      "Файлът със сертификата не е намерен"
+    IDS_CONTENTDB           "ContentDB (Уеб)"
+    IDS_DELETEAFTERFILEOP   "Файлът '%s' ще бъде изтрит след завършване на текущата операция"
+    IDS_DL_RESUMED          "Възобновяване на свалянето на ""%s"""
+    IDS_ED2KSERVER          "Сървър eD2K"
+    IDS_ENABLEIMPORTPARTS   "Разреши внасяне на части"
+    IDS_ERR_REGEXP          "Грешен регулярен израз"
+    IDS_IMPORTPARTS         "Внеси части..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Внасянето е прекратено: не може да се внася в завършен файл."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Внасянето е прекратено: %s не може да бъде отворен."
+    IDS_IMPORTPARTS_IMPORTSTART "Внасяне на %i части в %s."
+    IDS_IMPORTPARTS_STOP    "Спри внасянето на части"
+    IDS_IRC_BAN             "Забрани"
+    IDS_IRC_ENABLEUTF8      "Използвай UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Игнорирай съобщенията Ping? Pong!"
+    IDS_ITALIC              "Наклонен текст"
+    IDS_KEY_NOT_FOUND       "Файлът с ключа не е намерен"
+    IDS_PREVIEWREQUESTED    "Заявен е преглед - моля, изчакайте"
+    IDS_UNCADD              "Добави UNC път..."
+    IDS_WEB_CERT            "Сертификат"
+    IDS_WEB_GENERATE        "Създай нов сертификат"
+    IDS_WEB_HTTPS           "Използвай HTTPS"
 END
 
 #endif    // Bulgarian resources
diff --git a/srchybrid/lang/ca_ES.rc b/srchybrid/lang/ca_ES.rc
index d1ae131..5885d9e 100644
--- a/srchybrid/lang/ca_ES.rc
+++ b/srchybrid/lang/ca_ES.rc
@@ -1983,6 +1983,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Mètode d'autenticació"
+    IDS_BOOTSTRAPFAILED     "No s'ha pogut iniciar la Kad. El vostre fitxer nodes.dat pot ser obsolet o hi ha un problema de connexió."
+    IDS_CERT_CREATED        "S'ha creat un certificat nou; número de sèrie %d"
+    IDS_CERT_ERR_CREATE     "No s'ha creat el certificat (vegeu el registre Verbose)"
+    IDS_CERT_NOT_FOUND      "No s'ha trobat el fitxer del certificat"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "L'arxiu '%s' s'esborrarà quan acabi l'operació actual"
+    IDS_DL_RESUMED          "Es reprèn la descàrrega de ""%s"""
+    IDS_ED2KSERVER          "Servidor eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activar la importació de parts"
+    IDS_ERR_REGEXP          "Expressió regular incorrecta"
+    IDS_IMPORTPARTS         "Importar parts..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importació cancel·lada: no es pot importar a un arxiu complet."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importació cancel·lada: no s'ha pogut obrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Important %i parts a %s."
+    IDS_IMPORTPARTS_STOP    "Aturar la importació de parts"
+    IDS_IRC_BAN             "Bandejar"
+    IDS_IRC_ENABLEUTF8      "Activar UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignora els missatges Ping? Pong!"
+    IDS_ITALIC              "Commuta la cursiva"
+    IDS_KEY_NOT_FOUND       "No s'ha trobat el fitxer de la clau"
+    IDS_PREVIEWREQUESTED    "Previsualització sol·licitada - espereu"
+    IDS_UNCADD              "Afegir camí UNC..."
+    IDS_WEB_CERT            "Certificat"
+    IDS_WEB_GENERATE        "Crear un certificat nou"
+    IDS_WEB_HTTPS           "Utilitzar HTTPS"
 END
 
 #endif    // Catalan resources
diff --git a/srchybrid/lang/cz_CZ.rc b/srchybrid/lang/cz_CZ.rc
index b635e18..077ea6c 100644
--- a/srchybrid/lang/cz_CZ.rc
+++ b/srchybrid/lang/cz_CZ.rc
@@ -1979,6 +1979,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Povolit efekty hlavního panelu ve Windows 7"
     IDS_LANMODE             "LAN mód"
     IDS_OPENPREFINI         "Otevřít preferences.ini"
+    IDS_AUTHMETHOD          "Způsob ověření"
+    IDS_BOOTSTRAPFAILED     "Připojení ke Kad se nezdařilo. Váš soubor nodes.dat může být zastaralý nebo je problém s připojením."
+    IDS_CERT_CREATED        "Vytvořen nový certifikát; sériové číslo %d"
+    IDS_CERT_ERR_CREATE     "Certifikát nebyl vytvořen (viz Verbose log)"
+    IDS_CERT_NOT_FOUND      "Soubor certifikátu nenalezen"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Soubor '%s' bude smazán po dokončení současné operace"
+    IDS_DL_RESUMED          "Obnovuji stahování ""%s"""
+    IDS_ED2KSERVER          "eD2K server"
+    IDS_ENABLEIMPORTPARTS   "Povolit import částí"
+    IDS_ERR_REGEXP          "Chybný regulární výraz"
+    IDS_IMPORTPARTS         "Importovat části..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import přerušen: nelze importovat do kompletního souboru."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import přerušen: nelze otevřít %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importuji %i částí do %s."
+    IDS_IMPORTPARTS_STOP    "Zastavit import částí"
+    IDS_IRC_BAN             "Zakázat"
+    IDS_IRC_ENABLEUTF8      "Povolit UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorovat zprávy Ping? Pong!"
+    IDS_ITALIC              "Přepne kurzívu textu"
+    IDS_KEY_NOT_FOUND       "Soubor klíče nenalezen"
+    IDS_PREVIEWREQUESTED    "Náhled vyžádán - čekejte prosím"
+    IDS_UNCADD              "Přidat UNC cestu..."
+    IDS_WEB_CERT            "Certifikát"
+    IDS_WEB_GENERATE        "Vytvořit nový certifikát"
+    IDS_WEB_HTTPS           "Použít HTTPS"
 END
 
 #endif    // Czech resources
diff --git a/srchybrid/lang/da_DK.rc b/srchybrid/lang/da_DK.rc
index 5ea1107..1de7bf4 100644
--- a/srchybrid/lang/da_DK.rc
+++ b/srchybrid/lang/da_DK.rc
@@ -1987,6 +1987,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Godkendelsesmetode"
+    IDS_BOOTSTRAPFAILED     "Kunne ikke starte Kad. Din nodes.dat er måske forældet, eller der er et forbindelsesproblem."
+    IDS_CERT_CREATED        "Nyt certifikat oprettet; serienummer %d"
+    IDS_CERT_ERR_CREATE     "Certifikat ikke oprettet (se Verbose-loggen)"
+    IDS_CERT_NOT_FOUND      "Certifikatfilen blev ikke fundet"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Filen '%s' bliver slettet, når den aktuelle handling er afsluttet"
+    IDS_DL_RESUMED          "Genoptager download af ""%s"""
+    IDS_ED2KSERVER          "eD2K-server"
+    IDS_ENABLEIMPORTPARTS   "Aktiver import af dele"
+    IDS_ERR_REGEXP          "Ugyldigt regulært udtryk"
+    IDS_IMPORTPARTS         "Importer dele..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import afbrudt: Kan ikke importere til en komplet fil."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import afbrudt: Kunne ikke åbne %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importerer %i dele til %s."
+    IDS_IMPORTPARTS_STOP    "Stop import af dele"
+    IDS_IRC_BAN             "Udeluk"
+    IDS_IRC_ENABLEUTF8      "Aktiver UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorer Ping? Pong!-beskeder"
+    IDS_ITALIC              "Skift kursiv Tekst"
+    IDS_KEY_NOT_FOUND       "Nøglefilen blev ikke fundet"
+    IDS_PREVIEWREQUESTED    "Forhåndsvisning anmodet - vent venligst"
+    IDS_UNCADD              "Tilføj UNC-sti..."
+    IDS_WEB_CERT            "Certifikat"
+    IDS_WEB_GENERATE        "Opret nyt certifikat"
+    IDS_WEB_HTTPS           "Brug HTTPS"
 END
 
 #endif    // Danish resources
diff --git a/srchybrid/lang/de_DE.rc b/srchybrid/lang/de_DE.rc
index 60b4454..4bc50f7 100644
--- a/srchybrid/lang/de_DE.rc
+++ b/srchybrid/lang/de_DE.rc
@@ -1955,6 +1955,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Aktiviere Windows7 Taskbareffekte"
     IDS_LANMODE             "LAN Modus"
     IDS_OPENPREFINI         "Öffne preferences.ini"
+    IDS_AUTHMETHOD          "Authentifizierungsmethode"
+    IDS_BOOTSTRAPFAILED     "Kad-Bootstrap fehlgeschlagen. Die Datei nodes.dat ist eventuell veraltet oder es liegt ein Verbindungsproblem vor."
+    IDS_CERT_CREATED        "Neues Zertifikat erstellt; Seriennummer %d"
+    IDS_CERT_ERR_CREATE     "Zertifikat nicht erstellt (siehe Verbose-Log)"
+    IDS_CERT_NOT_FOUND      "Zertifikatsdatei nicht gefunden"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Die Datei '%s' wird nach Abschluss des aktuellen Vorgangs gelöscht"
+    IDS_DL_RESUMED          "Download von ""%s"" wird fortgesetzt"
+    IDS_ED2KSERVER          "eD2K-Server"
+    IDS_ENABLEIMPORTPARTS   "Teile-Import aktivieren"
+    IDS_ERR_REGEXP          "Ungültiger regulärer Ausdruck"
+    IDS_IMPORTPARTS         "Teile importieren..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import abgebrochen: In eine vollständige Datei kann nicht importiert werden."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import abgebrochen: %s konnte nicht geöffnet werden."
+    IDS_IMPORTPARTS_IMPORTSTART "Importiere %i Teile in %s."
+    IDS_IMPORTPARTS_STOP    "Teile-Import stoppen"
+    IDS_IRC_BAN             "Bannen"
+    IDS_IRC_ENABLEUTF8      "UTF-8 aktivieren"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoriere Ping?-Pong!-Nachrichten"
+    IDS_ITALIC              "Schaltet Kursivschrift an/aus"
+    IDS_KEY_NOT_FOUND       "Schlüsseldatei nicht gefunden"
+    IDS_PREVIEWREQUESTED    "Vorschau angefordert - bitte warten"
+    IDS_UNCADD              "UNC-Pfad hinzufügen..."
+    IDS_WEB_CERT            "Zertifikat"
+    IDS_WEB_GENERATE        "Neues Zertifikat erstellen"
+    IDS_WEB_HTTPS           "HTTPS verwenden"
 END
 
 
diff --git a/srchybrid/lang/el_GR.rc b/srchybrid/lang/el_GR.rc
index 5163c33..bb179ba 100644
--- a/srchybrid/lang/el_GR.rc
+++ b/srchybrid/lang/el_GR.rc
@@ -1922,6 +1922,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Ενεργοποίηση των Windows 7 taskbar effects"
     IDS_LANMODE             "Κατάσταση LAN"
     IDS_OPENPREFINI         "Άνοιγμα του preferences.ini"
+    IDS_AUTHMETHOD          "Μέθοδος πιστοποίησης"
+    IDS_BOOTSTRAPFAILED     "Αποτυχία σύνδεσης στο Kad. Το αρχείο nodes.dat μπορεί να είναι παλιό ή υπάρχει πρόβλημα σύνδεσης."
+    IDS_CERT_CREATED        "Δημιουργήθηκε νέο πιστοποιητικό, σειριακός αριθμός %d"
+    IDS_CERT_ERR_CREATE     "Το πιστοποιητικό δεν δημιουργήθηκε (δείτε το αρχείο καταγραφής Verbose)"
+    IDS_CERT_NOT_FOUND      "Το αρχείο πιστοποιητικού δεν βρέθηκε"
+    IDS_CONTENTDB           "ContentDB (Ιστός)"
+    IDS_DELETEAFTERFILEOP   "Το αρχείο '%s' θα διαγραφεί μόλις ολοκληρωθεί η τρέχουσα ενέργεια"
+    IDS_DL_RESUMED          "Συνέχιση κατεβάσματος του ""%s"""
+    IDS_ED2KSERVER          "Εξυπηρετητής eD2K"
+    IDS_ENABLEIMPORTPARTS   "Ενεργοποίηση εισαγωγής κομματιών"
+    IDS_ERR_REGEXP          "Λανθασμένη κανονική έκφραση"
+    IDS_IMPORTPARTS         "Εισαγωγή κομματιών..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Η εισαγωγή ακυρώθηκε: αδύνατη η εισαγωγή σε ολοκληρωμένο αρχείο."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Η εισαγωγή ακυρώθηκε: αδύνατο το άνοιγμα του %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Εισαγωγή %i κομματιών στο %s."
+    IDS_IMPORTPARTS_STOP    "Διακοπή εισαγωγής κομματιών"
+    IDS_IRC_BAN             "Αποκλεισμός"
+    IDS_IRC_ENABLEUTF8      "Ενεργοποίηση UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Αγνόηση μηνυμάτων Ping? Pong!"
+    IDS_ITALIC              "Εναλλαγή πλάγιου κειμένου"
+    IDS_KEY_NOT_FOUND       "Το αρχείο κλειδιού δεν βρέθηκε"
+    IDS_PREVIEWREQUESTED    "Ζητήθηκε προεπισκόπηση - παρακαλώ περιμένετε"
+    IDS_UNCADD              "Προσθήκη διαδρομής UNC..."
+    IDS_WEB_CERT            "Πιστοποιητικό"
+    IDS_WEB_GENERATE        "Δημιουργία νέου πιστοποιητικού"
+    IDS_WEB_HTTPS           "Χρήση HTTPS"
 END
 
 #endif    // Greek resources
diff --git a/srchybrid/lang/es_AS.rc b/srchybrid/lang/es_AS.rc
index a939aa9..2b32da5 100644
--- a/srchybrid/lang/es_AS.rc
+++ b/srchybrid/lang/es_AS.rc
@@ -1959,6 +1959,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Asturian  resources
diff --git a/srchybrid/lang/es_ES_T.rc b/srchybrid/lang/es_ES_T.rc
index 070f0c6..e22d52e 100644
--- a/srchybrid/lang/es_ES_T.rc
+++ b/srchybrid/lang/es_ES_T.rc
@@ -1982,6 +1982,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Activar efectos de la barra de tareas de Windows 7"
     IDS_LANMODE             "Modo LAN"
     IDS_OPENPREFINI         "Abrir preferences.ini"
+    IDS_AUTHMETHOD          "Método de autenticación"
+    IDS_BOOTSTRAPFAILED     "Fallo al arrancar Kad. Tu archivo nodes.dat puede estar obsoleto o hay un problema de conexión."
+    IDS_CERT_CREATED        "Nuevo certificado creado; número de serie %d"
+    IDS_CERT_ERR_CREATE     "Certificado no creado (ver registro Verbose)"
+    IDS_CERT_NOT_FOUND      "Archivo de certificado no encontrado"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "El archivo '%s' se borrará cuando termine la operación actual"
+    IDS_DL_RESUMED          "Reanudando la descarga de ""%s"""
+    IDS_ED2KSERVER          "Servidor eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activar la importación de partes"
+    IDS_ERR_REGEXP          "Expresión regular incorrecta"
+    IDS_IMPORTPARTS         "Importar partes..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importación cancelada: no se puede importar a un archivo completo."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importación cancelada: no se pudo abrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importando %i partes en %s."
+    IDS_IMPORTPARTS_STOP    "Detener la importación de partes"
+    IDS_IRC_BAN             "Vetar"
+    IDS_IRC_ENABLEUTF8      "Activar UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorar los mensajes Ping? Pong!"
+    IDS_ITALIC              "Cambia a Texto en Cursiva"
+    IDS_KEY_NOT_FOUND       "Archivo de clave no encontrado"
+    IDS_PREVIEWREQUESTED    "Vista previa solicitada - espera por favor"
+    IDS_UNCADD              "Añadir ruta UNC..."
+    IDS_WEB_CERT            "Certificado"
+    IDS_WEB_GENERATE        "Crear nuevo certificado"
+    IDS_WEB_HTTPS           "Usar HTTPS"
 END
 
 #endif    // Spanish (Castilian) resources
diff --git a/srchybrid/lang/et_EE.rc b/srchybrid/lang/et_EE.rc
index caa0999..434cf3e 100644
--- a/srchybrid/lang/et_EE.rc
+++ b/srchybrid/lang/et_EE.rc
@@ -1973,6 +1973,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Luba Windows 7 tööristariba effektid"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Ava seadistuste.ini"
+    IDS_AUTHMETHOD          "Autentimismeetod"
+    IDS_BOOTSTRAPFAILED     "Kad-võrguga liitumine ebaõnnestus. Fail nodes.dat võib olla aegunud või on ühendusega probleeme."
+    IDS_CERT_CREATED        "Uus sertifikaat loodud; seerianumber %d"
+    IDS_CERT_ERR_CREATE     "Sertifikaati ei loodud (vaata Verbose logi)"
+    IDS_CERT_NOT_FOUND      "Sertifikaadifaili ei leitud"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Fail '%s' kustutatakse pärast käimasoleva toimingu lõppu"
+    IDS_DL_RESUMED          "Jätkan faili ""%s"" tõmbamist"
+    IDS_ED2KSERVER          "eD2K server"
+    IDS_ENABLEIMPORTPARTS   "Luba osade import"
+    IDS_ERR_REGEXP          "Vigane regulaaravaldis"
+    IDS_IMPORTPARTS         "Impordi osi..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import katkestatud: valmis faili ei saa importida."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import katkestatud: faili %s ei õnnestunud avada."
+    IDS_IMPORTPARTS_IMPORTSTART "Impordin %i osa faili %s."
+    IDS_IMPORTPARTS_STOP    "Peata osade import"
+    IDS_IRC_BAN             "Bänni"
+    IDS_IRC_ENABLEUTF8      "Luba UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoreeri Ping? Pong! sõnumeid"
+    IDS_ITALIC              "Kaldkirjas teksti nupp"
+    IDS_KEY_NOT_FOUND       "Võtmefaili ei leitud"
+    IDS_PREVIEWREQUESTED    "Eelvaade tellitud - palun oota"
+    IDS_UNCADD              "Lisa UNC-tee..."
+    IDS_WEB_CERT            "Sertifikaat"
+    IDS_WEB_GENERATE        "Loo uus sertifikaat"
+    IDS_WEB_HTTPS           "Kasuta HTTPS-i"
 END
 
 #endif    // Estonian resources
diff --git a/srchybrid/lang/fa_IR.rc b/srchybrid/lang/fa_IR.rc
index ef61c75..5c1e838 100644
--- a/srchybrid/lang/fa_IR.rc
+++ b/srchybrid/lang/fa_IR.rc
@@ -1945,6 +1945,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "فعال کردن جلوه شيشه اي تسک بار ويندوز 7"
     IDS_LANMODE             "در حالت شبکه اتصال محلي LAN"
     IDS_OPENPREFINI         "باز کردن preferences.ini"
+    IDS_AUTHMETHOD          "روش احراز هويت"
+    IDS_BOOTSTRAPFAILED     "اتصال به شبكه كد ناموفق بود. فايل nodes.dat شما ممكن است قديمي باشد يا مشكل اتصال وجود دارد."
+    IDS_CERT_CREATED        "گواهي جديد ساخته شد; شماره سريال %d"
+    IDS_CERT_ERR_CREATE     "گواهي ساخته نشد (گزارش Verbose را ببينيد)"
+    IDS_CERT_NOT_FOUND      "فايل گواهي پيدا نشد"
+    IDS_CONTENTDB           "ContentDB (وب)"
+    IDS_DELETEAFTERFILEOP   "فايل '%s' پس از پايان عمليات جاري حذف خواهد شد"
+    IDS_DL_RESUMED          "از سرگيري دريافت ""%s"""
+    IDS_ED2KSERVER          "سرويس دهنده eD2K"
+    IDS_ENABLEIMPORTPARTS   "فعال کردن وارد کردن قطعه ها"
+    IDS_ERR_REGEXP          "عبارت باقاعده نادرست"
+    IDS_IMPORTPARTS         "وارد کردن قطعه ها..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "وارد کردن لغو شد: نمي توان به يك فايل كامل وارد كرد."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "وارد کردن لغو شد: %s باز نشد."
+    IDS_IMPORTPARTS_IMPORTSTART "وارد کردن %i قطعه به %s."
+    IDS_IMPORTPARTS_STOP    "توقف وارد کردن قطعه ها"
+    IDS_IRC_BAN             "ممنوع كردن"
+    IDS_IRC_ENABLEUTF8      "فعال کردن UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "پيغام هاي Ping? Pong! را ناديده بگير"
+    IDS_ITALIC              "سوييچ کردن نوشته کج شده"
+    IDS_KEY_NOT_FOUND       "فايل كليد پيدا نشد"
+    IDS_PREVIEWREQUESTED    "پيش نمايش درخواست شد - لطفا صبر كنيد"
+    IDS_UNCADD              "اضافه کردن مسير UNC..."
+    IDS_WEB_CERT            "گواهي"
+    IDS_WEB_GENERATE        "ساخت گواهي جديد"
+    IDS_WEB_HTTPS           "استفاده از HTTPS"
 END
 
 #endif
diff --git a/srchybrid/lang/fi_FI.rc b/srchybrid/lang/fi_FI.rc
index 289d9f9..2f76cab 100644
--- a/srchybrid/lang/fi_FI.rc
+++ b/srchybrid/lang/fi_FI.rc
@@ -1949,6 +1949,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Käytä Windows 7:n tehtäväpalkin efektejä"
     IDS_LANMODE             "LAN-tila"
     IDS_OPENPREFINI         "Avaa preferences.ini"
+    IDS_AUTHMETHOD          "Todennustapa"
+    IDS_BOOTSTRAPFAILED     "Kad-verkkoon liittyminen epäonnistui. Tiedosto nodes.dat voi olla vanhentunut tai yhteydessä on ongelma."
+    IDS_CERT_CREATED        "Uusi varmenne luotu; sarjanumero %d"
+    IDS_CERT_ERR_CREATE     "Varmennetta ei luotu (katso Verbose-loki)"
+    IDS_CERT_NOT_FOUND      "Varmennetiedostoa ei löydy"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Tiedosto '%s' poistetaan, kun nykyinen toiminto on valmis"
+    IDS_DL_RESUMED          "Jatketaan tiedoston ""%s"" latausta"
+    IDS_ED2KSERVER          "eD2K-palvelin"
+    IDS_ENABLEIMPORTPARTS   "Ota osien tuonti käyttöön"
+    IDS_ERR_REGEXP          "Virheellinen säännöllinen lauseke"
+    IDS_IMPORTPARTS         "Tuo osia..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Tuonti keskeytetty: valmiiseen tiedostoon ei voi tuoda."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Tuonti keskeytetty: tiedostoa %s ei voitu avata."
+    IDS_IMPORTPARTS_IMPORTSTART "Tuodaan %i osaa tiedostoon %s."
+    IDS_IMPORTPARTS_STOP    "Lopeta osien tuonti"
+    IDS_IRC_BAN             "Porttikielto"
+    IDS_IRC_ENABLEUTF8      "Ota UTF-8 käyttöön"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Suodata Ping? Pong! -viestit"
+    IDS_ITALIC              "Vaihtaa kursivoidun tekstin päälle/pois"
+    IDS_KEY_NOT_FOUND       "Avaintiedostoa ei löydy"
+    IDS_PREVIEWREQUESTED    "Esikatselua pyydetty - odota"
+    IDS_UNCADD              "Lisää UNC-polku..."
+    IDS_WEB_CERT            "Varmenne"
+    IDS_WEB_GENERATE        "Luo uusi varmenne"
+    IDS_WEB_HTTPS           "Käytä HTTPS:ää"
 END
 
 #endif    // Finnish resources
diff --git a/srchybrid/lang/fr_BR.rc b/srchybrid/lang/fr_BR.rc
index 772ecbe..f0ab683 100644
--- a/srchybrid/lang/fr_BR.rc
+++ b/srchybrid/lang/fr_BR.rc
@@ -1941,6 +1941,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//	IDS_AUTHMETHOD	"Authentication method"
+//	IDS_BOOTSTRAPFAILED	"Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//	IDS_CERT_CREATED	"New certificate created; serial %d"
+//	IDS_CERT_ERR_CREATE	"Certificate not created (see Verbose log)"
+//	IDS_CERT_NOT_FOUND	"Certificate file not found"
+//	IDS_CONTENTDB	"ContentDB (Web)"
+//	IDS_DELETEAFTERFILEOP	"File '%s' will be deleted after the current operation was completed"
+//	IDS_DL_RESUMED	"Resume downloading ""%s"""
+//	IDS_ED2KSERVER	"eD2K Server"
+//	IDS_ENABLEIMPORTPARTS	"Enable Import parts"
+//	IDS_ERR_REGEXP	"Bad regular expression"
+//	IDS_IMPORTPARTS	"Import parts..."
+//	IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE	"Import aborted: Can't import into a complete file."
+//	IDS_IMPORTPARTS_ERR_CANTOPENFILE	"Import aborted: Couldn't open %s."
+//	IDS_IMPORTPARTS_IMPORTSTART	"Importing %i parts into %s."
+//	IDS_IMPORTPARTS_STOP	"Stop Import parts"
+//	IDS_IRC_BAN	"Ban"
+//	IDS_IRC_ENABLEUTF8	"Enable UTF-8"
+//	IDS_IRC_IGNOREPINGPONGMESSAGE	"Ignore Ping? Pong! messages"
+//	IDS_ITALIC	"Toggles italicized text"
+//	IDS_KEY_NOT_FOUND	"Key file not found"
+//	IDS_PREVIEWREQUESTED	"Preview requested - please wait"
+//	IDS_UNCADD	"Add UNC path..."
+//	IDS_WEB_CERT	"Certificate"
+//	IDS_WEB_GENERATE	"Create new certificate"
+//	IDS_WEB_HTTPS	"Use HTTPS"
 END
 
 #endif    // Breton (France) resources
diff --git a/srchybrid/lang/fr_FR.rc b/srchybrid/lang/fr_FR.rc
index 3de9b41..211459c 100644
--- a/srchybrid/lang/fr_FR.rc
+++ b/srchybrid/lang/fr_FR.rc
@@ -1978,6 +1978,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Activer les effets Windows 7 de la barre des tâches"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Méthode d'authentification"
+    IDS_BOOTSTRAPFAILED     "Échec de l'amorçage de Kad. Votre fichier nodes.dat est peut-être obsolète ou il y a un problème de connexion."
+    IDS_CERT_CREATED        "Nouveau certificat créé ; numéro de série %d"
+    IDS_CERT_ERR_CREATE     "Certificat non créé (voir le journal Verbose)"
+    IDS_CERT_NOT_FOUND      "Fichier de certificat introuvable"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Le fichier '%s' sera supprimé à la fin de l'opération en cours"
+    IDS_DL_RESUMED          "Reprise du téléchargement de ""%s"""
+    IDS_ED2KSERVER          "Serveur eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activer l'import de parties"
+    IDS_ERR_REGEXP          "Expression régulière incorrecte"
+    IDS_IMPORTPARTS         "Importer des parties..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import annulé : impossible d'importer dans un fichier complet."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import annulé : impossible d'ouvrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Import de %i parties dans %s."
+    IDS_IMPORTPARTS_STOP    "Arrêter l'import de parties"
+    IDS_IRC_BAN             "Bannir"
+    IDS_IRC_ENABLEUTF8      "Activer l'UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorer les messages Ping? Pong!"
+    IDS_ITALIC              "Texte en italique"
+    IDS_KEY_NOT_FOUND       "Fichier de clé introuvable"
+    IDS_PREVIEWREQUESTED    "Aperçu demandé - veuillez patienter"
+    IDS_UNCADD              "Ajouter un chemin UNC..."
+    IDS_WEB_CERT            "Certificat"
+    IDS_WEB_GENERATE        "Créer un nouveau certificat"
+    IDS_WEB_HTTPS           "Utiliser HTTPS"
 END
 
 #endif    // French (France) resources
diff --git a/srchybrid/lang/gl_ES.rc b/srchybrid/lang/gl_ES.rc
index a096daa..73521cb 100644
--- a/srchybrid/lang/gl_ES.rc
+++ b/srchybrid/lang/gl_ES.rc
@@ -1986,6 +1986,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Activar efectos da barra de tarefas de Windows 7"
     IDS_LANMODE             "Modo LAN"
     IDS_OPENPREFINI         "Abrir preferences.ini"
+    IDS_AUTHMETHOD          "Método de autenticación"
+    IDS_BOOTSTRAPFAILED     "Fallo ao arrincar a Kad. O teu arquivo nodes.dat pode estar obsoleto ou hai un problema de conexión."
+    IDS_CERT_CREATED        "Novo certificado creado; número de serie %d"
+    IDS_CERT_ERR_CREATE     "Certificado non creado (ver rexistro Verbose)"
+    IDS_CERT_NOT_FOUND      "Arquivo de certificado non atopado"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "O arquivo '%s' borrarase cando remate a operación actual"
+    IDS_DL_RESUMED          "Retomando a descarga de ""%s"""
+    IDS_ED2KSERVER          "Servidor eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activar a importación de partes"
+    IDS_ERR_REGEXP          "Expresión regular incorrecta"
+    IDS_IMPORTPARTS         "Importar partes..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importación cancelada: non se pode importar a un arquivo completo."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importación cancelada: non se puido abrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importando %i partes en %s."
+    IDS_IMPORTPARTS_STOP    "Deter a importación de partes"
+    IDS_IRC_BAN             "Vetar"
+    IDS_IRC_ENABLEUTF8      "Activar UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorar as mensaxes Ping? Pong!"
+    IDS_ITALIC              "Cambia a texto en cursiva"
+    IDS_KEY_NOT_FOUND       "Arquivo de chave non atopado"
+    IDS_PREVIEWREQUESTED    "Vista previa solicitada - agarda por favor"
+    IDS_UNCADD              "Engadir ruta UNC..."
+    IDS_WEB_CERT            "Certificado"
+    IDS_WEB_GENERATE        "Crear novo certificado"
+    IDS_WEB_HTTPS           "Usar HTTPS"
 END
 
 #endif    // Galego: 0x56, 0x1 resources
diff --git a/srchybrid/lang/he_IL.rc b/srchybrid/lang/he_IL.rc
index bdaedc6..76caeca 100644
--- a/srchybrid/lang/he_IL.rc
+++ b/srchybrid/lang/he_IL.rc
@@ -1969,6 +1969,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "הפעל אפקטים של שורת מצב של חלונות 7"
     IDS_LANMODE             "מצב רשת מקומית (LAN)"
     IDS_OPENPREFINI         "פתח את preferences.ini"
+    IDS_AUTHMETHOD          "שיטת אימות"
+    IDS_BOOTSTRAPFAILED     "ההתחברות ל-Kad נכשלה. ייתכן שהקובץ nodes.dat מיושן או שיש בעיית חיבור."
+    IDS_CERT_CREATED        "נוצרה תעודה חדשה; מספר סידורי %d"
+    IDS_CERT_ERR_CREATE     "התעודה לא נוצרה (ראה יומן Verbose)"
+    IDS_CERT_NOT_FOUND      "קובץ התעודה לא נמצא"
+    IDS_CONTENTDB           "ContentDB (רשת)"
+    IDS_DELETEAFTERFILEOP   "הקובץ '%s' יימחק בתום הפעולה הנוכחית"
+    IDS_DL_RESUMED          "מחדש את הורדת ""%s"""
+    IDS_ED2KSERVER          "שרת eD2K"
+    IDS_ENABLEIMPORTPARTS   "הפעל ייבוא חלקים"
+    IDS_ERR_REGEXP          "ביטוי רגולרי שגוי"
+    IDS_IMPORTPARTS         "ייבוא חלקים..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "הייבוא בוטל: לא ניתן לייבא לקובץ שלם."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "הייבוא בוטל: לא ניתן לפתוח את %s."
+    IDS_IMPORTPARTS_IMPORTSTART "מייבא %i חלקים אל %s."
+    IDS_IMPORTPARTS_STOP    "עצור ייבוא חלקים"
+    IDS_IRC_BAN             "חסום"
+    IDS_IRC_ENABLEUTF8      "הפעל UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "התעלם מהודעות Ping? Pong!"
+    IDS_ITALIC              "החלף טקסט נטוי"
+    IDS_KEY_NOT_FOUND       "קובץ המפתח לא נמצא"
+    IDS_PREVIEWREQUESTED    "התבקשה תצוגה מקדימה - נא להמתין"
+    IDS_UNCADD              "הוסף נתיב UNC..."
+    IDS_WEB_CERT            "תעודה"
+    IDS_WEB_GENERATE        "צור תעודה חדשה"
+    IDS_WEB_HTTPS           "השתמש ב-HTTPS"
 END
 
 #endif    // Hebrew resources
diff --git a/srchybrid/lang/hu_HU.rc b/srchybrid/lang/hu_HU.rc
index 156411b..d3d25ad 100644
--- a/srchybrid/lang/hu_HU.rc
+++ b/srchybrid/lang/hu_HU.rc
@@ -1983,6 +1983,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Hitelesítési mód"
+    IDS_BOOTSTRAPFAILED     "A Kad indítása nem sikerült. A nodes.dat fájlod elavult lehet, vagy kapcsolódási hiba történt."
+    IDS_CERT_CREATED        "Új tanúsítvány létrehozva; sorozatszám: %d"
+    IDS_CERT_ERR_CREATE     "A tanúsítvány nem jött létre (lásd a Verbose naplót)"
+    IDS_CERT_NOT_FOUND      "A tanúsítványfájl nem található"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "A(z) '%s' fájl törlésre kerül a folyamatban lévő művelet befejezése után"
+    IDS_DL_RESUMED          "A(z) ""%s"" letöltésének folytatása"
+    IDS_ED2KSERVER          "eD2K kiszolgáló"
+    IDS_ENABLEIMPORTPARTS   "Darabok importálásának engedélyezése"
+    IDS_ERR_REGEXP          "Hibás reguláris kifejezés"
+    IDS_IMPORTPARTS         "Darabok importálása..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Az importálás megszakadt: kész fájlba nem lehet importálni."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Az importálás megszakadt: a(z) %s nem nyitható meg."
+    IDS_IMPORTPARTS_IMPORTSTART "%i darab importálása ide: %s."
+    IDS_IMPORTPARTS_STOP    "Darabok importálásának leállítása"
+    IDS_IRC_BAN             "Kitilt"
+    IDS_IRC_ENABLEUTF8      "UTF-8 engedélyezése"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ping? Pong! üzenetek mellőzése"
+    IDS_ITALIC              "Dőlt szöveg be/ki"
+    IDS_KEY_NOT_FOUND       "A kulcsfájl nem található"
+    IDS_PREVIEWREQUESTED    "Előnézet kérve - kérlek várj"
+    IDS_UNCADD              "UNC útvonal hozzáadása..."
+    IDS_WEB_CERT            "Tanúsítvány"
+    IDS_WEB_GENERATE        "Új tanúsítvány létrehozása"
+    IDS_WEB_HTTPS           "HTTPS használata"
 END
 
 #endif    // Hungarian resources
diff --git a/srchybrid/lang/it_IT.rc b/srchybrid/lang/it_IT.rc
index 973aa82..79bfec4 100644
--- a/srchybrid/lang/it_IT.rc
+++ b/srchybrid/lang/it_IT.rc
@@ -1980,6 +1980,9 @@ BEGIN
     IDS_IRC_ENABLEUTF8      "Abilita UTF-8"
     IDS_ITALIC              "Testo corsivo"
 //    IDS_IRC_BAN             "Ban"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_ED2KSERVER          "Server eD2K"
+    IDS_IRC_BAN             "Bandisci"
 END
 
 #endif    // Italian (Italy) resources
diff --git a/srchybrid/lang/jp_JP.rc b/srchybrid/lang/jp_JP.rc
index 7bad0a9..225bd4e 100644
--- a/srchybrid/lang/jp_JP.rc
+++ b/srchybrid/lang/jp_JP.rc
@@ -1983,6 +1983,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "認証方式"
+    IDS_BOOTSTRAPFAILED     "Kadへの接続に失敗しました。nodes.datが古いか、接続に問題があります。"
+    IDS_CERT_CREATED        "新しい証明書を作成しました。シリアル番号 %d"
+    IDS_CERT_ERR_CREATE     "証明書を作成できませんでした (Verboseログを参照)"
+    IDS_CERT_NOT_FOUND      "証明書ファイルが見つかりません"
+    IDS_CONTENTDB           "ContentDB (ウェブ)"
+    IDS_DELETEAFTERFILEOP   "ファイル '%s' は現在の処理が完了した後に削除されます"
+    IDS_DL_RESUMED          """%s"" のダウンロードを再開します"
+    IDS_ED2KSERVER          "eD2Kサーバー"
+    IDS_ENABLEIMPORTPARTS   "パートのインポートを有効にする"
+    IDS_ERR_REGEXP          "正規表現が不正です"
+    IDS_IMPORTPARTS         "パートのインポート..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "インポートを中止しました: 完了したファイルにはインポートできません。"
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "インポートを中止しました: %s を開けませんでした。"
+    IDS_IMPORTPARTS_IMPORTSTART "%i 個のパートを %s にインポートしています。"
+    IDS_IMPORTPARTS_STOP    "パートのインポートを停止"
+    IDS_IRC_BAN             "アクセス禁止"
+    IDS_IRC_ENABLEUTF8      "UTF-8を有効にする"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ping? Pong! メッセージを無視"
+    IDS_ITALIC              "テキストを斜体にする"
+    IDS_KEY_NOT_FOUND       "鍵ファイルが見つかりません"
+    IDS_PREVIEWREQUESTED    "プレビューを要求しました - お待ちください"
+    IDS_UNCADD              "UNCパスを追加..."
+    IDS_WEB_CERT            "証明書"
+    IDS_WEB_GENERATE        "新しい証明書を作成"
+    IDS_WEB_HTTPS           "HTTPSを使用する"
 END
 
 #endif    // 日本語 (Japanese) resources
diff --git a/srchybrid/lang/ko_KR.rc b/srchybrid/lang/ko_KR.rc
index 842d3d1..167d6e2 100644
--- a/srchybrid/lang/ko_KR.rc
+++ b/srchybrid/lang/ko_KR.rc
@@ -1943,6 +1943,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "인증 방식"
+    IDS_BOOTSTRAPFAILED     "Kad 접속에 실패했습니다. nodes.dat 파일이 오래되었거나 연결에 문제가 있습니다."
+    IDS_CERT_CREATED        "새 인증서를 만들었습니다. 일련번호 %d"
+    IDS_CERT_ERR_CREATE     "인증서를 만들지 못했습니다 (Verbose 로그 참조)"
+    IDS_CERT_NOT_FOUND      "인증서 파일을 찾을 수 없음"
+    IDS_CONTENTDB           "ContentDB (웹)"
+    IDS_DELETEAFTERFILEOP   "'%s' 파일은 현재 작업이 끝난 후 삭제됩니다"
+    IDS_DL_RESUMED          """%s"" 다운로드를 다시 시작합니다"
+    IDS_ED2KSERVER          "eD2K 서버"
+    IDS_ENABLEIMPORTPARTS   "조각 가져오기 사용"
+    IDS_ERR_REGEXP          "잘못된 정규 표현식"
+    IDS_IMPORTPARTS         "조각 가져오기..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "가져오기 취소: 완료된 파일에는 가져올 수 없습니다."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "가져오기 취소: %s 을(를) 열 수 없습니다."
+    IDS_IMPORTPARTS_IMPORTSTART "%i 개의 조각을 %s 에 가져오는 중입니다."
+    IDS_IMPORTPARTS_STOP    "조각 가져오기 중지"
+    IDS_IRC_BAN             "차단"
+    IDS_IRC_ENABLEUTF8      "UTF-8 사용"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ping? Pong! 메시지 무시"
+    IDS_ITALIC              "기울임 글꼴"
+    IDS_KEY_NOT_FOUND       "키 파일을 찾을 수 없음"
+    IDS_PREVIEWREQUESTED    "미리보기를 요청했습니다 - 잠시 기다려 주십시오"
+    IDS_UNCADD              "UNC 경로 추가..."
+    IDS_WEB_CERT            "인증서"
+    IDS_WEB_GENERATE        "새 인증서 만들기"
+    IDS_WEB_HTTPS           "HTTPS 사용"
 END
 
 #endif    // Korean resources
diff --git a/srchybrid/lang/lt_LT.rc b/srchybrid/lang/lt_LT.rc
index 3960233..7dd784b 100644
--- a/srchybrid/lang/lt_LT.rc
+++ b/srchybrid/lang/lt_LT.rc
@@ -1980,6 +1980,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Įjungti Windows 7 užduočių juostos efektus"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Tapatybės nustatymo būdas"
+    IDS_BOOTSTRAPFAILED     "Nepavyko prisijungti prie Kad. Jūsų nodes.dat gali būti pasenęs arba yra ryšio problema."
+    IDS_CERT_CREATED        "Sukurtas naujas sertifikatas; serijos numeris %d"
+    IDS_CERT_ERR_CREATE     "Sertifikatas nesukurtas (žiūrėkite Verbose žurnalą)"
+    IDS_CERT_NOT_FOUND      "Sertifikato failas nerastas"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Failas '%s' bus ištrintas pasibaigus dabartinei operacijai"
+    IDS_DL_RESUMED          "Tęsiamas ""%s"" siuntimas"
+    IDS_ED2KSERVER          "eD2K serveris"
+    IDS_ENABLEIMPORTPARTS   "Įjungti dalių importavimą"
+    IDS_ERR_REGEXP          "Neteisinga reguliarioji išraiška"
+    IDS_IMPORTPARTS         "Importuoti dalis..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importavimas nutrauktas: negalima importuoti į baigtą failą."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importavimas nutrauktas: nepavyko atverti %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importuojamos %i dalys į %s."
+    IDS_IMPORTPARTS_STOP    "Sustabdyti dalių importavimą"
+    IDS_IRC_BAN             "Užblokuoti"
+    IDS_IRC_ENABLEUTF8      "Įjungti UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoruoti Ping? Pong! pranešimus"
+    IDS_ITALIC              "Įjungti kursyvinį tekstą"
+    IDS_KEY_NOT_FOUND       "Rakto failas nerastas"
+    IDS_PREVIEWREQUESTED    "Peržiūra užsakyta - prašome palaukti"
+    IDS_UNCADD              "Pridėti UNC kelią..."
+    IDS_WEB_CERT            "Sertifikatas"
+    IDS_WEB_GENERATE        "Sukurti naują sertifikatą"
+    IDS_WEB_HTTPS           "Naudoti HTTPS"
 END
 
 #endif    // Lithuanian resources
diff --git a/srchybrid/lang/lv_LV.rc b/srchybrid/lang/lv_LV.rc
index 14147ad..54398c7 100644
--- a/srchybrid/lang/lv_LV.rc
+++ b/srchybrid/lang/lv_LV.rc
@@ -1974,6 +1974,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Latvian resources
diff --git a/srchybrid/lang/mt_MT.rc b/srchybrid/lang/mt_MT.rc
index b77f122..01555d4 100644
--- a/srchybrid/lang/mt_MT.rc
+++ b/srchybrid/lang/mt_MT.rc
@@ -1970,6 +1970,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Maltese resources
diff --git a/srchybrid/lang/nb_NO.rc b/srchybrid/lang/nb_NO.rc
index 75f0e3d..e919273 100644
--- a/srchybrid/lang/nb_NO.rc
+++ b/srchybrid/lang/nb_NO.rc
@@ -1980,6 +1980,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Autentiseringsmetode"
+    IDS_BOOTSTRAPFAILED     "Kunne ikke starte Kad. Din nodes.dat kan være utdatert, eller det er et tilkoblingsproblem."
+    IDS_CERT_CREATED        "Nytt sertifikat opprettet; serienummer %d"
+    IDS_CERT_ERR_CREATE     "Sertifikatet ble ikke opprettet (se Verbose-loggen)"
+    IDS_CERT_NOT_FOUND      "Sertifikatfilen ble ikke funnet"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Filen '%s' vil bli slettet når den pågående operasjonen er fullført"
+    IDS_DL_RESUMED          "Gjenopptar nedlasting av ""%s"""
+    IDS_ED2KSERVER          "eD2K-tjener"
+    IDS_ENABLEIMPORTPARTS   "Aktiver import av deler"
+    IDS_ERR_REGEXP          "Ugyldig regulært uttrykk"
+    IDS_IMPORTPARTS         "Importer deler..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import avbrutt: Kan ikke importere til en komplett fil."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import avbrutt: Kunne ikke åpne %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importerer %i deler til %s."
+    IDS_IMPORTPARTS_STOP    "Stopp import av deler"
+    IDS_IRC_BAN             "Utesteng"
+    IDS_IRC_ENABLEUTF8      "Aktiver UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorer Ping? Pong!-meldinger"
+    IDS_ITALIC              "Skifter kursiv tekst"
+    IDS_KEY_NOT_FOUND       "Nøkkelfilen ble ikke funnet"
+    IDS_PREVIEWREQUESTED    "Forhåndsvisning forespurt - vennligst vent"
+    IDS_UNCADD              "Legg til UNC-sti..."
+    IDS_WEB_CERT            "Sertifikat"
+    IDS_WEB_GENERATE        "Opprett nytt sertifikat"
+    IDS_WEB_HTTPS           "Bruk HTTPS"
 END
 
 #endif    // Norwegian (Bokmal) resources
diff --git a/srchybrid/lang/nl_NL.rc b/srchybrid/lang/nl_NL.rc
index 71e8fe3..bb598ae 100644
--- a/srchybrid/lang/nl_NL.rc
+++ b/srchybrid/lang/nl_NL.rc
@@ -1980,6 +1980,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Taakbalkeffecten van Windows 7 inschakelen"
     IDS_LANMODE             "LAN-mode"
     IDS_OPENPREFINI         "preferences.ini openen"
+    IDS_AUTHMETHOD          "Authenticatiemethode"
+    IDS_BOOTSTRAPFAILED     "Bootstrappen van Kad mislukt. Uw nodes.dat is mogelijk verouderd of er is een verbindingsprobleem."
+    IDS_CERT_CREATED        "Nieuw certificaat aangemaakt; serienummer %d"
+    IDS_CERT_ERR_CREATE     "Certificaat niet aangemaakt (zie Verbose-log)"
+    IDS_CERT_NOT_FOUND      "Certificaatbestand niet gevonden"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Het bestand '%s' wordt verwijderd zodra de huidige bewerking klaar is"
+    IDS_DL_RESUMED          "Download van ""%s"" wordt hervat"
+    IDS_ED2KSERVER          "eD2K-server"
+    IDS_ENABLEIMPORTPARTS   "Importeren van delen inschakelen"
+    IDS_ERR_REGEXP          "Ongeldige reguliere expressie"
+    IDS_IMPORTPARTS         "Delen importeren..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import afgebroken: kan niet importeren in een compleet bestand."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import afgebroken: kon %s niet openen."
+    IDS_IMPORTPARTS_IMPORTSTART "Bezig met importeren van %i delen in %s."
+    IDS_IMPORTPARTS_STOP    "Importeren van delen stoppen"
+    IDS_IRC_BAN             "Bannen"
+    IDS_IRC_ENABLEUTF8      "UTF-8 inschakelen"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Negeer Ping? Pong!-berichten"
+    IDS_ITALIC              "Maakt de tekst cursief"
+    IDS_KEY_NOT_FOUND       "Sleutelbestand niet gevonden"
+    IDS_PREVIEWREQUESTED    "Voorvertoning aangevraagd - even geduld"
+    IDS_UNCADD              "UNC-pad toevoegen..."
+    IDS_WEB_CERT            "Certificaat"
+    IDS_WEB_GENERATE        "Nieuw certificaat aanmaken"
+    IDS_WEB_HTTPS           "HTTPS gebruiken"
 END
 
 #endif    // Dutch (Netherlands) resources
diff --git a/srchybrid/lang/nn_NO.rc b/srchybrid/lang/nn_NO.rc
index 38dc709..624697d 100644
--- a/srchybrid/lang/nn_NO.rc
+++ b/srchybrid/lang/nn_NO.rc
@@ -1933,6 +1933,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Autentiseringsmetode"
+    IDS_BOOTSTRAPFAILED     "Kunne ikkje starte Kad. Fila nodes.dat kan vere utdatert, eller det er eit tilkoplingsproblem."
+    IDS_CERT_CREATED        "Nytt sertifikat oppretta; serienummer %d"
+    IDS_CERT_ERR_CREATE     "Sertifikatet vart ikkje oppretta (sjå Verbose-loggen)"
+    IDS_CERT_NOT_FOUND      "Fann ikkje sertifikatfila"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Fila '%s' vert sletta når den pågåande operasjonen er ferdig"
+    IDS_DL_RESUMED          "Held fram nedlastinga av ""%s"""
+    IDS_ED2KSERVER          "eD2K-tenar"
+    IDS_ENABLEIMPORTPARTS   "Aktiver import av delar"
+    IDS_ERR_REGEXP          "Ugyldig regulært uttrykk"
+    IDS_IMPORTPARTS         "Importer delar..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import avbroten: Kan ikkje importere til ei komplett fil."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import avbroten: Kunne ikkje opne %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importerer %i delar til %s."
+    IDS_IMPORTPARTS_STOP    "Stopp import av delar"
+    IDS_IRC_BAN             "Utesteng"
+    IDS_IRC_ENABLEUTF8      "Aktiver UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorer Ping? Pong!-meldingar"
+    IDS_ITALIC              "Set kursiv tekst på/av"
+    IDS_KEY_NOT_FOUND       "Fann ikkje nøkkelfila"
+    IDS_PREVIEWREQUESTED    "Førehandsvising etterspurd - vent litt"
+    IDS_UNCADD              "Legg til UNC-sti..."
+    IDS_WEB_CERT            "Sertifikat"
+    IDS_WEB_GENERATE        "Opprett nytt sertifikat"
+    IDS_WEB_HTTPS           "Bruk HTTPS"
 END
 
 #endif    // Norwegian (Nynorsk) resources
diff --git a/srchybrid/lang/pl_PL.rc b/srchybrid/lang/pl_PL.rc
index 6508acf..edd4c3c 100644
--- a/srchybrid/lang/pl_PL.rc
+++ b/srchybrid/lang/pl_PL.rc
@@ -1954,6 +1954,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Włącz efekty pasku zadań Windows 7"
     IDS_LANMODE             "Tryb LAN"
     IDS_OPENPREFINI         "Otwórz preferences.ini"
+    IDS_AUTHMETHOD          "Metoda uwierzytelniania"
+    IDS_BOOTSTRAPFAILED     "Nie udało się podłączyć do Kad. Twój plik nodes.dat może być nieaktualny lub występuje problem z połączeniem."
+    IDS_CERT_CREATED        "Utworzono nowy certyfikat; numer seryjny %d"
+    IDS_CERT_ERR_CREATE     "Nie utworzono certyfikatu (zobacz log Verbose)"
+    IDS_CERT_NOT_FOUND      "Nie znaleziono pliku certyfikatu"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Plik '%s' zostanie usunięty po zakończeniu bieżącej operacji"
+    IDS_DL_RESUMED          "Wznawianie pobierania ""%s"""
+    IDS_ED2KSERVER          "Serwer eD2K"
+    IDS_ENABLEIMPORTPARTS   "Włącz import części"
+    IDS_ERR_REGEXP          "Błędne wyrażenie regularne"
+    IDS_IMPORTPARTS         "Importuj części..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import przerwany: nie można importować do ukończonego pliku."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import przerwany: nie można otworzyć %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importowanie %i części do %s."
+    IDS_IMPORTPARTS_STOP    "Zatrzymaj import części"
+    IDS_IRC_BAN             "Zbanuj"
+    IDS_IRC_ENABLEUTF8      "Włącz UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoruj wiadomości Ping? Pong!"
+    IDS_ITALIC              "Włącz kursywę"
+    IDS_KEY_NOT_FOUND       "Nie znaleziono pliku klucza"
+    IDS_PREVIEWREQUESTED    "Zażądano podglądu - proszę czekać"
+    IDS_UNCADD              "Dodaj ścieżkę UNC..."
+    IDS_WEB_CERT            "Certyfikat"
+    IDS_WEB_GENERATE        "Utwórz nowy certyfikat"
+    IDS_WEB_HTTPS           "Używaj HTTPS"
 END
 
 #endif    // Polish resources
diff --git a/srchybrid/lang/pt_BR.rc b/srchybrid/lang/pt_BR.rc
index a94b369..281d29c 100644
--- a/srchybrid/lang/pt_BR.rc
+++ b/srchybrid/lang/pt_BR.rc
@@ -1972,6 +1972,23 @@ STRINGTABLE
 BEGIN
     IDS_WEB_GENERATE        "Criar novo certificado"
     IDS_CERT_ERR_CREATE     "Certificado não criado (veja Verbose log)"
+    IDS_AUTHMETHOD          "Método de autenticação"
+    IDS_BOOTSTRAPFAILED     "Falha ao iniciar a Kad. Seu arquivo nodes.dat pode estar desatualizado ou há um problema de conexão."
+    IDS_CERT_CREATED        "Novo certificado criado; número de série %d"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "O arquivo '%s' será apagado após a conclusão da operação atual"
+    IDS_DL_RESUMED          "Retomando o download de ""%s"""
+    IDS_ED2KSERVER          "Servidor eD2K"
+    IDS_ENABLEIMPORTPARTS   "Habilitar a importação de partes"
+    IDS_ERR_REGEXP          "Expressão regular inválida"
+    IDS_IMPORTPARTS         "Importar partes..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importação cancelada: não é possível importar para um arquivo completo."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importação cancelada: não foi possível abrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importando %i partes para %s."
+    IDS_IMPORTPARTS_STOP    "Parar a importação de partes"
+    IDS_IRC_BAN             "Banir"
+    IDS_PREVIEWREQUESTED    "Pré-visualização solicitada - aguarde"
+    IDS_UNCADD              "Adicionar caminho UNC..."
 END
 
 #endif    // Portuguese (Brazil) resources
diff --git a/srchybrid/lang/pt_PT.rc b/srchybrid/lang/pt_PT.rc
index 65ac274..b682694 100644
--- a/srchybrid/lang/pt_PT.rc
+++ b/srchybrid/lang/pt_PT.rc
@@ -1987,6 +1987,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Método de autenticação"
+    IDS_BOOTSTRAPFAILED     "Falha ao arrancar a Kad. O seu ficheiro nodes.dat pode estar desactualizado ou existe um problema de ligação."
+    IDS_CERT_CREATED        "Novo certificado criado; número de série %d"
+    IDS_CERT_ERR_CREATE     "Certificado não criado (ver registo Verbose)"
+    IDS_CERT_NOT_FOUND      "Ficheiro de certificado não encontrado"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "O ficheiro '%s' será apagado depois de concluída a operação actual"
+    IDS_DL_RESUMED          "A retomar a transferência de ""%s"""
+    IDS_ED2KSERVER          "Servidor eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activar a importação de partes"
+    IDS_ERR_REGEXP          "Expressão regular inválida"
+    IDS_IMPORTPARTS         "Importar partes..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importação cancelada: não é possível importar para um ficheiro completo."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importação cancelada: não foi possível abrir %s."
+    IDS_IMPORTPARTS_IMPORTSTART "A importar %i partes para %s."
+    IDS_IMPORTPARTS_STOP    "Parar a importação de partes"
+    IDS_IRC_BAN             "Banir"
+    IDS_IRC_ENABLEUTF8      "Activar UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorar mensagens Ping? Pong!"
+    IDS_ITALIC              "Alterna o Texto a Itálico"
+    IDS_KEY_NOT_FOUND       "Ficheiro de chave não encontrado"
+    IDS_PREVIEWREQUESTED    "Pré-visualização pedida - aguarde"
+    IDS_UNCADD              "Adicionar caminho UNC..."
+    IDS_WEB_CERT            "Certificado"
+    IDS_WEB_GENERATE        "Criar novo certificado"
+    IDS_WEB_HTTPS           "Usar HTTPS"
 END
 
 #endif    // Portuguese (Portugal) resources
diff --git a/srchybrid/lang/ro_RO.rc b/srchybrid/lang/ro_RO.rc
index 8f5ebf4..e0ed011 100644
--- a/srchybrid/lang/ro_RO.rc
+++ b/srchybrid/lang/ro_RO.rc
@@ -1982,6 +1982,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Activare efecte taskbar Windows 7"
     IDS_LANMODE             "Mod LAN"
     IDS_OPENPREFINI         "Deschide preferences.ini"
+    IDS_AUTHMETHOD          "Metodă de autentificare"
+    IDS_BOOTSTRAPFAILED     "Conectarea la Kad a eşuat. Fişierul nodes.dat poate fi învechit sau există o problemă de conexiune."
+    IDS_CERT_CREATED        "Certificat nou creat; serie %d"
+    IDS_CERT_ERR_CREATE     "Certificatul nu a fost creat (vezi jurnalul Verbose)"
+    IDS_CERT_NOT_FOUND      "Fişierul certificat nu a fost găsit"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Fişierul '%s' va fi şters după terminarea operaţiei curente"
+    IDS_DL_RESUMED          "Se reia descărcarea ""%s"""
+    IDS_ED2KSERVER          "Server eD2K"
+    IDS_ENABLEIMPORTPARTS   "Activează importul de părţi"
+    IDS_ERR_REGEXP          "Expresie regulată incorectă"
+    IDS_IMPORTPARTS         "Importă părţi..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import anulat: nu se poate importa într-un fişier complet."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import anulat: nu am putut deschide %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Import %i părţi în %s."
+    IDS_IMPORTPARTS_STOP    "Opreşte importul de părţi"
+    IDS_IRC_BAN             "Interzice"
+    IDS_IRC_ENABLEUTF8      "Activează UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoră mesajele Ping? Pong!"
+    IDS_ITALIC              "Text cursiv da/nu"
+    IDS_KEY_NOT_FOUND       "Fişierul cheie nu a fost găsit"
+    IDS_PREVIEWREQUESTED    "Previzualizare cerută - aşteptaţi"
+    IDS_UNCADD              "Adaugă cale UNC..."
+    IDS_WEB_CERT            "Certificat"
+    IDS_WEB_GENERATE        "Creează un certificat nou"
+    IDS_WEB_HTTPS           "Foloseşte HTTPS"
 END
 
 /////////////////////////////////////////////////////////////////////////////
diff --git a/srchybrid/lang/ru_RU.rc b/srchybrid/lang/ru_RU.rc
index cccfe9d..9aea9da 100644
--- a/srchybrid/lang/ru_RU.rc
+++ b/srchybrid/lang/ru_RU.rc
@@ -1980,6 +1980,8 @@ BEGIN
     IDS_IRC_ENABLEUTF8      "Включить UTF-8"
     IDS_ITALIC              "Переключение курсива"
 //    IDS_IRC_BAN             "Ban"
+    IDS_CONTENTDB           "ContentDB (Веб)"
+    IDS_IRC_BAN             "Забанить"
 END
 
 #endif    // Russian resources
diff --git a/srchybrid/lang/sl_SI.rc b/srchybrid/lang/sl_SI.rc
index 8301017..42d944b 100644
--- a/srchybrid/lang/sl_SI.rc
+++ b/srchybrid/lang/sl_SI.rc
@@ -1978,6 +1978,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Vklopi efekte orodne vrstice v Windows 7"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Način overjanja"
+    IDS_BOOTSTRAPFAILED     "Zagon omrežja Kad ni uspel. Vaša datoteka nodes.dat je morda zastarela ali pa gre za težavo s povezavo."
+    IDS_CERT_CREATED        "Ustvarjeno novo potrdilo; serijska številka %d"
+    IDS_CERT_ERR_CREATE     "Potrdilo ni bilo ustvarjeno (glejte dnevnik Verbose)"
+    IDS_CERT_NOT_FOUND      "Datoteke potrdila ni mogoče najti"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Datoteka '%s' bo izbrisana, ko se trenutna operacija zaključi"
+    IDS_DL_RESUMED          "Nadaljevanje prenosa ""%s"""
+    IDS_ED2KSERVER          "Strežnik eD2K"
+    IDS_ENABLEIMPORTPARTS   "Omogoči uvoz delov"
+    IDS_ERR_REGEXP          "Neveljaven regularni izraz"
+    IDS_IMPORTPARTS         "Uvozi dele..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Uvoz prekinjen: v dokončano datoteko ni mogoče uvažati."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Uvoz prekinjen: ni bilo mogoče odpreti %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Uvažanje %i delov v %s."
+    IDS_IMPORTPARTS_STOP    "Ustavi uvoz delov"
+    IDS_IRC_BAN             "Prepovej"
+    IDS_IRC_ENABLEUTF8      "Omogoči UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignoriraj sporočila Ping? Pong!"
+    IDS_ITALIC              "Preklopi na poševni tekst"
+    IDS_KEY_NOT_FOUND       "Datoteke s ključem ni mogoče najti"
+    IDS_PREVIEWREQUESTED    "Predogled zahtevan - prosimo, počakajte"
+    IDS_UNCADD              "Dodaj pot UNC..."
+    IDS_WEB_CERT            "Potrdilo"
+    IDS_WEB_GENERATE        "Ustvari novo potrdilo"
+    IDS_WEB_HTTPS           "Uporabi HTTPS"
 END
 
 #endif    // Slovenian resources
diff --git a/srchybrid/lang/sq_AL.rc b/srchybrid/lang/sq_AL.rc
index 0c00047..1252011 100644
--- a/srchybrid/lang/sq_AL.rc
+++ b/srchybrid/lang/sq_AL.rc
@@ -1973,6 +1973,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Metoda e vërtetimit"
+    IDS_BOOTSTRAPFAILED     "Nisja e Kad dështoi. Skeda juaj nodes.dat mund të jetë e vjetruar ose ka një problem lidhjeje."
+    IDS_CERT_CREATED        "U krijua një certifikatë e re; numri serial %d"
+    IDS_CERT_ERR_CREATE     "Certifikata nuk u krijua (shiko regjistrin Verbose)"
+    IDS_CERT_NOT_FOUND      "Skeda e certifikatës nuk u gjet"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Skeda '%s' do të fshihet pasi të përfundojë veprimi aktual"
+    IDS_DL_RESUMED          "Rifillon shkarkimi i ""%s"""
+    IDS_ED2KSERVER          "Serveri eD2K"
+    IDS_ENABLEIMPORTPARTS   "Lejo importimin e pjesëve"
+    IDS_ERR_REGEXP          "Shprehje e rregullt e gabuar"
+    IDS_IMPORTPARTS         "Importo pjesë..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importimi u ndërpre: nuk mund të importohet në një skedë të plotë."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importimi u ndërpre: nuk u hap dot %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Duke importuar %i pjesë në %s."
+    IDS_IMPORTPARTS_STOP    "Ndalo importimin e pjesëve"
+    IDS_IRC_BAN             "Dëbo"
+    IDS_IRC_ENABLEUTF8      "Lejo UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Përbuz mesazhet Ping? Pong!"
+    IDS_ITALIC              "Vë Tekstin e Pjerrët"
+    IDS_KEY_NOT_FOUND       "Skeda e çelësit nuk u gjet"
+    IDS_PREVIEWREQUESTED    "Parashikimi u kërkua - ju lutemi prisni"
+    IDS_UNCADD              "Shto shteg UNC..."
+    IDS_WEB_CERT            "Certifikata"
+    IDS_WEB_GENERATE        "Krijo një certifikatë të re"
+    IDS_WEB_HTTPS           "Përdor HTTPS"
 END
 
 #endif    // Albanisch resources
diff --git a/srchybrid/lang/sv_SE.rc b/srchybrid/lang/sv_SE.rc
index c808967..2d7a983 100644
--- a/srchybrid/lang/sv_SE.rc
+++ b/srchybrid/lang/sv_SE.rc
@@ -1980,6 +1980,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Autentiseringsmetod"
+    IDS_BOOTSTRAPFAILED     "Kunde inte starta Kad. Din nodes.dat kan vara föråldrad eller så finns det ett anslutningsproblem."
+    IDS_CERT_CREATED        "Nytt certifikat skapat; serienummer %d"
+    IDS_CERT_ERR_CREATE     "Certifikatet skapades inte (se Verbose-loggen)"
+    IDS_CERT_NOT_FOUND      "Certifikatfilen hittades inte"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Filen '%s' kommer att raderas när den pågående åtgärden är klar"
+    IDS_DL_RESUMED          "Återupptar hämtningen av ""%s"""
+    IDS_ED2KSERVER          "eD2K-server"
+    IDS_ENABLEIMPORTPARTS   "Aktivera import av delar"
+    IDS_ERR_REGEXP          "Ogiltigt reguljärt uttryck"
+    IDS_IMPORTPARTS         "Importera delar..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Importen avbröts: det går inte att importera till en färdig fil."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Importen avbröts: kunde inte öppna %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Importerar %i delar till %s."
+    IDS_IMPORTPARTS_STOP    "Stoppa import av delar"
+    IDS_IRC_BAN             "Bannlys"
+    IDS_IRC_ENABLEUTF8      "Aktivera UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignorera Ping? Pong!-meddelanden"
+    IDS_ITALIC              "Kursiv text"
+    IDS_KEY_NOT_FOUND       "Nyckelfilen hittades inte"
+    IDS_PREVIEWREQUESTED    "Förhandsgranskning begärd - vänta"
+    IDS_UNCADD              "Lägg till UNC-sökväg..."
+    IDS_WEB_CERT            "Certifikat"
+    IDS_WEB_GENERATE        "Skapa nytt certifikat"
+    IDS_WEB_HTTPS           "Använd HTTPS"
 END
 
 #endif    // Swedish resources
diff --git a/srchybrid/lang/tr_TR.rc b/srchybrid/lang/tr_TR.rc
index 1651999..65e7669 100644
--- a/srchybrid/lang/tr_TR.rc
+++ b/srchybrid/lang/tr_TR.rc
@@ -1979,6 +1979,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Windows 7 görev çubuğu efektleri etkin"
     IDS_LANMODE             "LAN Kipi"
     IDS_OPENPREFINI         "Preferences.ini'yi aç"
+    IDS_AUTHMETHOD          "Kimlik doğrulama yöntemi"
+    IDS_BOOTSTRAPFAILED     "Kad başlatılamadı. nodes.dat dosyanız eski olabilir veya bir bağlantı sorunu var."
+    IDS_CERT_CREATED        "Yeni sertifika oluşturuldu; seri numarası %d"
+    IDS_CERT_ERR_CREATE     "Sertifika oluşturulmadı (Verbose günlüğüne bakın)"
+    IDS_CERT_NOT_FOUND      "Sertifika dosyası bulunamadı"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "'%s' dosyası geçerli işlem tamamlandıktan sonra silinecek"
+    IDS_DL_RESUMED          """%s"" indirmesi sürdürülüyor"
+    IDS_ED2KSERVER          "eD2K Sunucusu"
+    IDS_ENABLEIMPORTPARTS   "Parça içe aktarma etkin"
+    IDS_ERR_REGEXP          "Hatalı düzenli ifade"
+    IDS_IMPORTPARTS         "Parçaları içe aktar..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "İçe aktarma iptal edildi: tamamlanmış bir dosyaya aktarılamaz."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "İçe aktarma iptal edildi: %s açılamadı."
+    IDS_IMPORTPARTS_IMPORTSTART "%i parça %s dosyasına aktarılıyor."
+    IDS_IMPORTPARTS_STOP    "Parça içe aktarmayı durdur"
+    IDS_IRC_BAN             "Yasakla"
+    IDS_IRC_ENABLEUTF8      "UTF-8 etkin"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ping? Pong! mesajlarını yoksay"
+    IDS_ITALIC              "Eğik metine değiştirir"
+    IDS_KEY_NOT_FOUND       "Anahtar dosyası bulunamadı"
+    IDS_PREVIEWREQUESTED    "Önizleme istendi - lütfen bekleyin"
+    IDS_UNCADD              "UNC yolu ekle..."
+    IDS_WEB_CERT            "Sertifika"
+    IDS_WEB_GENERATE        "Yeni sertifika oluştur"
+    IDS_WEB_HTTPS           "HTTPS kullan"
 END
 
 #endif    // Turkish resources
diff --git a/srchybrid/lang/ua_UA.rc b/srchybrid/lang/ua_UA.rc
index 0ad202d..0713bb3 100644
--- a/srchybrid/lang/ua_UA.rc
+++ b/srchybrid/lang/ua_UA.rc
@@ -1973,6 +1973,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+    IDS_AUTHMETHOD          "Спосіб автентифікації"
+    IDS_BOOTSTRAPFAILED     "Не вдалося під'єднатися до Kad. Ваш файл nodes.dat може бути застарілим або є проблема зі з'єднанням."
+    IDS_CERT_CREATED        "Створено новий сертифікат; серійний номер %d"
+    IDS_CERT_ERR_CREATE     "Сертифікат не створено (див. журнал Verbose)"
+    IDS_CERT_NOT_FOUND      "Файл сертифіката не знайдено"
+    IDS_CONTENTDB           "ContentDB (Веб)"
+    IDS_DELETEAFTERFILEOP   "Файл '%s' буде видалено після завершення поточної операції"
+    IDS_DL_RESUMED          "Відновлення завантаження ""%s"""
+    IDS_ED2KSERVER          "Сервер eD2K"
+    IDS_ENABLEIMPORTPARTS   "Дозволити імпорт частин"
+    IDS_ERR_REGEXP          "Помилка в регулярному виразі"
+    IDS_IMPORTPARTS         "Імпортувати частини..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Імпорт скасовано: неможливо імпортувати до завершеного файлу."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Імпорт скасовано: неможливо відкрити %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Імпортовано %i частин до %s."
+    IDS_IMPORTPARTS_STOP    "Припинити імпорт частин"
+    IDS_IRC_BAN             "Заблокувати"
+    IDS_IRC_ENABLEUTF8      "Увімкнути UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Ігнорувати повідомлення Ping? Pong!"
+    IDS_ITALIC              "Курсивний текст"
+    IDS_KEY_NOT_FOUND       "Файл ключа не знайдено"
+    IDS_PREVIEWREQUESTED    "Попередній перегляд запитано - зачекайте"
+    IDS_UNCADD              "Додати шлях UNC..."
+    IDS_WEB_CERT            "Сертифікат"
+    IDS_WEB_GENERATE        "Створити новий сертифікат"
+    IDS_WEB_HTTPS           "Використовувати HTTPS"
 END
 
 #endif    // Ukrainian resources
diff --git a/srchybrid/lang/ug_CN.rc b/srchybrid/lang/ug_CN.rc
index 03c9956..a83c98a 100644
--- a/srchybrid/lang/ug_CN.rc
+++ b/srchybrid/lang/ug_CN.rc
@@ -1944,6 +1944,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "‫Windows 7 ۋەزىپە بالداق ئۈنۈمىنى قوزغات‬"
     IDS_LANMODE             "دائىرىلىك تور ھالىتى"
     IDS_OPENPREFINI         "‫سەپلىمە ھۆججەت preferences.ini نى ئاچ‬"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Uyghur (PRC) resources
diff --git a/srchybrid/lang/va_ES.rc b/srchybrid/lang/va_ES.rc
index 61a5559..c58b5f8 100644
--- a/srchybrid/lang/va_ES.rc
+++ b/srchybrid/lang/va_ES.rc
@@ -1983,6 +1983,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Catalan resources
diff --git a/srchybrid/lang/va_ES_RACV.rc b/srchybrid/lang/va_ES_RACV.rc
index 9e790ae..2d24894 100644
--- a/srchybrid/lang/va_ES_RACV.rc
+++ b/srchybrid/lang/va_ES_RACV.rc
@@ -1984,6 +1984,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Enable Windows 7 taskbar effects"
     IDS_LANMODE             "LAN Mode"
     IDS_OPENPREFINI         "Open preferences.ini"
+//    IDS_AUTHMETHOD          "Authentication method"
+//    IDS_BOOTSTRAPFAILED     "Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+//    IDS_CERT_CREATED        "New certificate created; serial %d"
+//    IDS_CERT_ERR_CREATE     "Certificate not created (see Verbose log)"
+//    IDS_CERT_NOT_FOUND      "Certificate file not found"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+//    IDS_DELETEAFTERFILEOP   "File '%s' will be deleted after the current operation was completed"
+//    IDS_DL_RESUMED          "Resume downloading ""%s"""
+//    IDS_ED2KSERVER          "eD2K Server"
+//    IDS_ENABLEIMPORTPARTS   "Enable Import parts"
+//    IDS_ERR_REGEXP          "Bad regular expression"
+//    IDS_IMPORTPARTS         "Import parts..."
+//    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Import aborted: Can't import into a complete file."
+//    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Import aborted: Couldn't open %s."
+//    IDS_IMPORTPARTS_IMPORTSTART "Importing %i parts into %s."
+//    IDS_IMPORTPARTS_STOP    "Stop Import parts"
+//    IDS_IRC_BAN             "Ban"
+//    IDS_IRC_ENABLEUTF8      "Enable UTF-8"
+//    IDS_IRC_IGNOREPINGPONGMESSAGE "Ignore Ping? Pong! messages"
+//    IDS_ITALIC              "Toggles italicized text"
+//    IDS_KEY_NOT_FOUND       "Key file not found"
+//    IDS_PREVIEWREQUESTED    "Preview requested - please wait"
+//    IDS_UNCADD              "Add UNC path..."
+//    IDS_WEB_CERT            "Certificate"
+//    IDS_WEB_GENERATE        "Create new certificate"
+//    IDS_WEB_HTTPS           "Use HTTPS"
 END
 
 #endif    // Valencian (RACV) resources
diff --git a/srchybrid/lang/vi_VN.rc b/srchybrid/lang/vi_VN.rc
index 585ecfe..a9ca89e 100644
--- a/srchybrid/lang/vi_VN.rc
+++ b/srchybrid/lang/vi_VN.rc
@@ -1974,6 +1974,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "Cho phép hiệu ứng thanh công cụ Windows 7"
     IDS_LANMODE             "Chế độ Mạng nội bộ"
     IDS_OPENPREFINI         "Mở preferences.ini"
+    IDS_AUTHMETHOD          "Phương thức xác thực"
+    IDS_BOOTSTRAPFAILED     "Không thể khởi động Kad. Tập tin nodes.dat của bạn có thể đã cũ hoặc có sự cố kết nối."
+    IDS_CERT_CREATED        "Đã tạo chứng chỉ mới; số sê-ri %d"
+    IDS_CERT_ERR_CREATE     "Chứng chỉ chưa được tạo (xem nhật ký Verbose)"
+    IDS_CERT_NOT_FOUND      "Không tìm thấy tập tin chứng chỉ"
+//    IDS_CONTENTDB           "ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "Tập tin '%s' sẽ bị xóa sau khi thao tác hiện tại hoàn tất"
+    IDS_DL_RESUMED          "Tiếp tục tải xuống ""%s"""
+    IDS_ED2KSERVER          "Máy chủ eD2K"
+    IDS_ENABLEIMPORTPARTS   "Kích hoạt nhập các phần"
+    IDS_ERR_REGEXP          "Biểu thức chính quy không hợp lệ"
+    IDS_IMPORTPARTS         "Nhập các phần..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "Đã hủy nhập: không thể nhập vào một tập tin đã hoàn tất."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "Đã hủy nhập: không thể mở %s."
+    IDS_IMPORTPARTS_IMPORTSTART "Đang nhập %i phần vào %s."
+    IDS_IMPORTPARTS_STOP    "Dừng nhập các phần"
+    IDS_IRC_BAN             "Cấm"
+    IDS_IRC_ENABLEUTF8      "Kích hoạt UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "Bỏ qua thông điệp Ping? Pong!"
+    IDS_ITALIC              "Bật chữ nghiêng"
+    IDS_KEY_NOT_FOUND       "Không tìm thấy tập tin khóa"
+    IDS_PREVIEWREQUESTED    "Đã yêu cầu xem trước - vui lòng đợi"
+    IDS_UNCADD              "Thêm đường dẫn UNC..."
+    IDS_WEB_CERT            "Chứng chỉ"
+    IDS_WEB_GENERATE        "Tạo chứng chỉ mới"
+    IDS_WEB_HTTPS           "Sử dụng HTTPS"
 END
 
 #endif    // Vietnamese resources
diff --git a/srchybrid/lang/zh_CN.rc b/srchybrid/lang/zh_CN.rc
index bb115f7..d5e4a0a 100644
--- a/srchybrid/lang/zh_CN.rc
+++ b/srchybrid/lang/zh_CN.rc
@@ -1954,6 +1954,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "启用 Winodws 7 任务栏特效" //"Enable Windows 7 taskbar effects"
     IDS_LANMODE             "局域网模式" //"LAN Mode"
     IDS_OPENPREFINI         "打开配置文件(preferences.ini)" //"Open preferences.ini"
+    IDS_AUTHMETHOD          "验证方式" //"Authentication method"
+    IDS_BOOTSTRAPFAILED     "连接 Kad 网络失败. 你的 nodes.dat 可能已过期, 或者连接有问题." //"Failed to bootstrap Kad. Your nodes.dat might be outdated or there is a connection problem."
+    IDS_CERT_CREATED        "已创建新证书; 序列号 %d" //"New certificate created; serial %d"
+    IDS_CERT_ERR_CREATE     "证书未创建 (请查看 Verbose 日志)" //"Certificate not created (see Verbose log)"
+    IDS_CERT_NOT_FOUND      "找不到证书文件" //"Certificate file not found"
+    IDS_CONTENTDB           "ContentDB (网络)" //"ContentDB (Web)"
+    IDS_DELETEAFTERFILEOP   "文件 '%s' 将在当前操作完成后删除" //"File '%s' will be deleted after the current operation was completed"
+    IDS_DL_RESUMED          "继续下载 ""%s""" //"Resume downloading ""%s"""
+    IDS_ED2KSERVER          "eD2K 服务器" //"eD2K Server"
+    IDS_ENABLEIMPORTPARTS   "启用分块导入" //"Enable Import parts"
+    IDS_ERR_REGEXP          "正则表达式错误" //"Bad regular expression"
+    IDS_IMPORTPARTS         "导入分块..." //"Import parts..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "导入中止: 无法导入到已完成的文件." //"Import aborted: Can't import into a complete file."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "导入中止: 无法打开 %s." //"Import aborted: Couldn't open %s."
+    IDS_IMPORTPARTS_IMPORTSTART "正在将 %i 个分块导入到 %s." //"Importing %i parts into %s."
+    IDS_IMPORTPARTS_STOP    "停止分块导入" //"Stop Import parts"
+    IDS_IRC_BAN             "封禁" //"Ban"
+    IDS_IRC_ENABLEUTF8      "启用 UTF-8" //"Enable UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "忽略 Ping? Pong! 消息" //"Ignore Ping? Pong! messages"
+    IDS_ITALIC              "切换到斜体字" //"Toggles italicized text"
+    IDS_KEY_NOT_FOUND       "找不到密钥文件" //"Key file not found"
+    IDS_PREVIEWREQUESTED    "已请求预览 - 请稍候" //"Preview requested - please wait"
+    IDS_UNCADD              "添加 UNC 路径..." //"Add UNC path..."
+    IDS_WEB_CERT            "证书" //"Certificate"
+    IDS_WEB_GENERATE        "创建新证书" //"Create new certificate"
+    IDS_WEB_HTTPS           "使用 HTTPS" //"Use HTTPS"
 END
 
 
diff --git a/srchybrid/lang/zh_TW.rc b/srchybrid/lang/zh_TW.rc
index 5578355..dce09a6 100644
--- a/srchybrid/lang/zh_TW.rc
+++ b/srchybrid/lang/zh_TW.rc
@@ -1954,6 +1954,32 @@ BEGIN
     IDS_SHOWWIN7TASKBARGOODIES "啟用 Windows 7 工作列效果"
     IDS_LANMODE             "LAN 模式"
     IDS_OPENPREFINI         "開啟 preferences.ini"
+    IDS_AUTHMETHOD          "驗證方式"
+    IDS_BOOTSTRAPFAILED     "連線至 Kad 失敗. 您的 nodes.dat 可能已過期, 或是連線有問題."
+    IDS_CERT_CREATED        "已建立新憑證; 序號 %d"
+    IDS_CERT_ERR_CREATE     "憑證未建立 (請查看 Verbose 紀錄)"
+    IDS_CERT_NOT_FOUND      "找不到憑證檔案"
+    IDS_CONTENTDB           "ContentDB (網路)"
+    IDS_DELETEAFTERFILEOP   "檔案 '%s' 將在目前的作業完成後刪除"
+    IDS_DL_RESUMED          "繼續下載 ""%s"""
+    IDS_ED2KSERVER          "eD2K 伺服器"
+    IDS_ENABLEIMPORTPARTS   "啟用部分檔案匯入"
+    IDS_ERR_REGEXP          "不正確的正規表示式"
+    IDS_IMPORTPARTS         "匯入部分檔案..."
+    IDS_IMPORTPARTS_ERR_ALREADYCOMPLETE "匯入中止: 無法匯入到已完成的檔案."
+    IDS_IMPORTPARTS_ERR_CANTOPENFILE "匯入中止: 無法開啟 %s."
+    IDS_IMPORTPARTS_IMPORTSTART "正在將 %i 個部分匯入到 %s."
+    IDS_IMPORTPARTS_STOP    "停止部分檔案匯入"
+    IDS_IRC_BAN             "封鎖"
+    IDS_IRC_ENABLEUTF8      "啟用 UTF-8"
+    IDS_IRC_IGNOREPINGPONGMESSAGE "忽略 Ping? Pong! 訊息"
+    IDS_ITALIC              "固定為斜體文字"
+    IDS_KEY_NOT_FOUND       "找不到金鑰檔案"
+    IDS_PREVIEWREQUESTED    "已要求預覽 - 請稍候"
+    IDS_UNCADD              "新增 UNC 路徑..."
+    IDS_WEB_CERT            "憑證"
+    IDS_WEB_GENERATE        "建立新憑證"
+    IDS_WEB_HTTPS           "使用 HTTPS"
 END
 
 /////////////////////////////////////////////////////////////////////////////
