diff --git a/srchybrid/Stdafx.h b/srchybrid/Stdafx.h
index 4e980f9..075d99e 100644
--- a/srchybrid/Stdafx.h
+++ b/srchybrid/Stdafx.h
@@ -98,10 +98,10 @@
 #pragma warning(disable:4264) // <func>: no override available for virtual member function from base <class>; function is hidden
 #pragma warning(disable:4265) // <class>: class has virtual functions, but destructor is not virtual
 #if _MSC_VER>=1400
-#pragma warning(disable:4266) // no override available for virtual member function from base <class>; function is hidden
+//#pragma warning(disable:4266) // no override available for virtual member function from base <class>; function is hidden
 #endif
 #if _MSC_VER>=1400
-#pragma warning(disable:4365) // conversion from 'int' to 'UINT', signed/unsigned mismatch
+//#pragma warning(disable:4365) // conversion from 'int' to 'UINT', signed/unsigned mismatch
 #endif
 #if _MSC_VER>=1900
 #pragma warning(disable:4435) // Object layout under /vd2 will change due to virtual base
